<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2023-12-13" />

<title>multi-tissue analysis (LD merge =T)</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">ctwas_Tibetan</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xsun1229/ctwas_Tibetan">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">multi-tissue analysis (LD merge =T)</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2023-12-13</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-12-17
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>ctwas_Tibetan/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20231213code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20231213)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20231213code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20231213)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229ctwasTibetantree60ba183938bf13d5e3137a47f5b2661fdf51b1b2targetblank60ba183a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/xsun1229/ctwas_Tibetan/tree/60ba183938bf13d5e3137a47f5b2661fdf51b1b2" target="_blank">60ba183</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229ctwasTibetantree60ba183938bf13d5e3137a47f5b2661fdf51b1b2targetblank60ba183a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/xsun1229/ctwas_Tibetan/tree/60ba183938bf13d5e3137a47f5b2661fdf51b1b2" target="_blank">60ba183</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    ctwas_Tibetan/.Rhistory

Untracked files:
    Untracked:  .Rhistory
    Untracked:  factor_analysis/
    Untracked:  factor_analysis_DGN/
    Untracked:  factor_analysis_new/
    Untracked:  heart_atlas/
    Untracked:  heart_project/
    Untracked:  module_analysis/
    Untracked:  multigroup_ctwas_analysis/
    Untracked:  mutation_rate/
    Untracked:  pathway_analysis/
    Untracked:  proteome_alzheimer/

Unstaged changes:
    Modified:   ctwas_Tibetan/analysis/multi_tissue.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>ctwas_Tibetan/analysis/multi_tissue.Rmd</code>) and HTML
(<code>ctwas_Tibetan/docs/multi_tissue.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/ctwas_Tibetan/blob/60ba183938bf13d5e3137a47f5b2661fdf51b1b2/ctwas_Tibetan/analysis/multi_tissue.Rmd" target="_blank">60ba183</a>
</td>
<td>
XSun
</td>
<td>
2023-12-14
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>For each trait, we sorted the tissues by PVE explained by genes, then
the top5 un-correlated tissues were selected for the multi-tissue
analysis.</p>
<p><a
href="https://xsun1229.github.io/ctwas_Tibetan/uncorrelated_tissue.html">link
to deciding tissues</a></p>
<div id="data" class="section level1">
<h1>Data</h1>
<div id="traits-and-selected-tissues" class="section level2">
<h2>Traits and selected tissues</h2>
<table>
<thead>
<tr class="header">
<th>Trait</th>
<th>Tissue</th>
<th>PVE_gene</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>LB</td>
<td>Lung</td>
<td>0.0410350101026084</td>
</tr>
<tr class="even">
<td></td>
<td>Whole_Blood</td>
<td>0.0110875627240866</td>
</tr>
<tr class="odd">
<td></td>
<td>Heart_Left_Ventricle</td>
<td>0.0107697033082494</td>
</tr>
<tr class="even">
<td></td>
<td>Artery_Coronary</td>
<td>0.00930578096193273</td>
</tr>
<tr class="odd">
<td></td>
<td>Kidney_Cortex</td>
<td>0.00426258259876705</td>
</tr>
<tr class="even">
<td>MC</td>
<td>Liver</td>
<td>0.0178648723821638</td>
</tr>
<tr class="odd">
<td></td>
<td>Artery_Tibial</td>
<td>0.0134706822007496</td>
</tr>
<tr class="even">
<td></td>
<td>Heart_Atrial_Appendage</td>
<td>0.0100733429644444</td>
</tr>
<tr class="odd">
<td></td>
<td>Pancreas</td>
<td>0.00867082220578753</td>
</tr>
<tr class="even">
<td></td>
<td>Lung</td>
<td>0.0085924869917463</td>
</tr>
<tr class="odd">
<td>OxHb</td>
<td>Heart_Left_Ventricle</td>
<td>0.0633139032762666</td>
</tr>
<tr class="even">
<td></td>
<td>Kidney_Cortex</td>
<td>0.0397523122784917</td>
</tr>
<tr class="odd">
<td></td>
<td>Liver</td>
<td>0.0231916966396772</td>
</tr>
<tr class="even">
<td></td>
<td>Artery_Coronary</td>
<td>0.0135241915935851</td>
</tr>
<tr class="odd">
<td></td>
<td>Lung</td>
<td>0.0104129270975456</td>
</tr>
<tr class="even">
<td>Pulse</td>
<td>Heart_Atrial_Appendage</td>
<td>0.0492054125707622</td>
</tr>
<tr class="odd">
<td></td>
<td>Kidney_Cortex</td>
<td>0.0402428509363196</td>
</tr>
<tr class="even">
<td></td>
<td>Artery_Tibial</td>
<td>0.0173924354650824</td>
</tr>
<tr class="odd">
<td></td>
<td>Whole_Blood</td>
<td>0.0148621621982472</td>
</tr>
<tr class="even">
<td></td>
<td>Liver</td>
<td>0.00959987160023217</td>
</tr>
</tbody>
</table>
</div>
<div id="functions-used" class="section level2">
<h2>Functions used</h2>
<pre class="r"><code>suppressMessages(library(ctwas))
suppressMessages(library(data.table))
suppressMessages(library(RSQLite))
suppressMessages(library(DT))
suppressMessages(library(tools))
suppressMessages(library(dplyr))
suppressMessages(library(tidyr))

thin &lt;- 0.1

process_data &lt;- function(weight, outputdir, outname, z_snp, z_gene, display_datatable = TRUE) {
  # Splitting and constructing weight file paths
  weight &lt;- unlist(strsplit(weight, split=&quot;,&quot;))
  weight &lt;- paste0(&quot;/project2/xinhe/shared_data/multigroup_ctwas/weights/mashr_models/mashr_&quot;, weight, &quot;.db&quot;)

  # Reading cTWAS results
  ctwas_res &lt;- fread(paste0(outputdir, outname, &quot;.susieIrss.txt&quot;))

  # Initialize gene_info dataframe
  gene_info &lt;- data.frame(gene=character(), genename=character(), gene_type=character(), weight=character())
  for (i in seq_along(weight)) {
    sqlite &lt;- dbDriver(&quot;SQLite&quot;)
    db &lt;- dbConnect(sqlite, weight[i])
    query &lt;- function(...) dbGetQuery(db, ...)
    gene_info_current &lt;- query(&quot;SELECT gene, genename, gene_type FROM extra&quot;)
    dbDisconnect(db)

    gene_info_current$weight &lt;- weight[i]
    gene_info &lt;- rbind(gene_info, gene_info_current)
  }

  # Processing gene_info
  gene_info$weight &lt;- sapply(gene_info$weight, function(x) rev(unlist(strsplit(file_path_sans_ext(x), &quot;/&quot;)))[1])
  gene_info$id &lt;- paste(gene_info$gene, gene_info$weight, sep=&quot;|&quot;)

  # Adding gene names to cTWAS results
  ctwas_res$genename[ctwas_res$type != &quot;SNP&quot;] &lt;- gene_info$genename[match(ctwas_res$id[ctwas_res$type != &quot;SNP&quot;], gene_info$id)]

  # Adding z values
  ctwas_res$z[ctwas_res$type == &quot;SNP&quot;] &lt;- z_snp$z[match(ctwas_res$id[ctwas_res$type == &quot;SNP&quot;], z_snp$id)]
  ctwas_res$z[ctwas_res$type != &quot;SNP&quot;] &lt;- z_gene$z[match(ctwas_res$id[ctwas_res$type != &quot;SNP&quot;], z_gene$id)]

  # Sorting and displaying datatable if required
  ctwas_res &lt;- ctwas_res[order(-ctwas_res$susie_pip),]
  if (display_datatable) {
    datatable(ctwas_res[ctwas_res$type != &quot;SNP&quot; &amp; ctwas_res$susie_pip &gt; 0.8,])
  }
  
  return(ctwas_res)
}

process_gene_data &lt;- function(ctwas_res, weight = NULL) {
  # Aggregate susie_pip
  df_gene &lt;- aggregate(ctwas_res$susie_pip[ctwas_res$type != &quot;SNP&quot;], 
                       by = list(ctwas_res$genename[ctwas_res$type != &quot;SNP&quot;]), 
                       FUN = sum)
  colnames(df_gene) &lt;- c(&quot;genename&quot;, &quot;combined_pip&quot;)

  # Optional part for dropping duplicated gene names
  # Uncomment this part if needed
  # df_gene &lt;- df_gene[!(df_gene$genename %in% names(which(table(ctwas_res$genename) &gt; length(weight)))),]

  # Collect tissue-level results
  all_tissue_names &lt;- unique(ctwas_res$type[ctwas_res$type != &quot;SNP&quot;])

  df_gene_pips &lt;- matrix(NA, nrow = nrow(df_gene), ncol = length(all_tissue_names))
  colnames(df_gene_pips) &lt;- all_tissue_names

  ctwas_gene_res &lt;- ctwas_res[ctwas_res$type != &quot;SNP&quot;,]

  for (i in 1:nrow(df_gene)) {
    gene &lt;- df_gene$genename[i]
    ctwas_gene_res_subset &lt;- ctwas_gene_res[ctwas_gene_res$genename == gene,]
    df_gene_pips[i, ctwas_gene_res_subset$type] &lt;- ctwas_gene_res_subset$susie_pip
  }

  df_gene &lt;- cbind(df_gene, df_gene_pips)

  # Sort by combined PIP
  df_gene &lt;- df_gene[order(-df_gene$combined_pip),]
  df_gene &lt;- df_gene[, apply(df_gene, 2, function(x) {!all(is.na(x))})]  # Drop genes not imputed in any tissue

  # Determine number of rows to display
  n_rows &lt;- max(sum(df_gene$combined_pip &gt; 0.8), 20)

  # Select top rows based on the criteria
  df_to_display &lt;- head(df_gene, n_rows)
  
  # Return a datatable
 # return(DT::datatable(df_to_display,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Genes with combined pip &gt; 0.8, 0 means pip=0; NA means no such gene in this tissue&#39;),options = list(pageLength = 10) ))
    datatable_display &lt;- DT::datatable(df_to_display, 
                                     caption = htmltools::tags$caption(style = &#39;caption-side: left; text-align: left; color:black; font-size:150%;&#39;, 
                                     &#39;Genes with combined pip &gt; 0.8, 0 means pip=0; NA means no such gene in this tissue&#39;), 
                                     options = list(pageLength = 10))
  return(df_to_display)
}

process_ctwas_results &lt;- function(ctwas_parameters, ctwas_res) {
    # Process parameters
    para &lt;- cbind(ctwas_parameters$group_size,
                  ctwas_parameters$group_prior,
                  ctwas_parameters$group_prior_var,
                  c(NA, ctwas_parameters$enrichment),
                  ctwas_parameters$group_pve)
    para &lt;- as.data.frame(cbind(rownames(para), para))
    colnames(para) &lt;- c(&quot;group&quot;,&quot;group_size&quot;,&quot;group_prior&quot;,&quot;group_prior_var&quot;,&quot;enrichment&quot;,&quot;group_pve&quot;)

    # Filter ctwas results for genes and high PIP
    ctwas_res_gene &lt;- ctwas_res[ctwas_res$type != &quot;SNP&quot;,]
    
    max_susie_pip_by_type &lt;- ctwas_res_gene %&gt;%
    group_by(type) %&gt;%
    summarize(max_susie_pip = max(susie_pip, na.rm = TRUE))

    # Merge summary with parameters
    summary &lt;- merge(max_susie_pip_by_type, para, by.x = &quot;type&quot;, by.y = &quot;group&quot;, all.y = TRUE)

    return(summary)
}

attributable_pip &lt;- function(df, combined_pip_col) {
  # Identify the relevant columns (those ending with &#39;_nolnc&#39;)
  #relevant_cols &lt;- names(df)[grepl(&quot;_nolnc$&quot;, names(df))]
  relevant_cols &lt;- names(df)[!names(df) %in%c( &quot;combined_pip&quot;,&quot;genename&quot;)]
  # Create and rearrange columns dynamically
  df &lt;- df %&gt;%
    mutate(across(all_of(relevant_cols), 
                  ~ ifelse(!is.na(.x) &amp; !is.na(combined_pip), 
                           .x / combined_pip * 100, NA), 
                  .names = &quot;{.col}_Attributable_PIP (%)&quot;))%&gt;%
    {
      # Building the new column order
      new_order &lt;- unlist(lapply(relevant_cols, function(col) {
        c(col, paste0(col, &quot;_Attributable_PIP (%)&quot;))
      }))
      select(., all_of(c(&quot;genename&quot;, combined_pip_col, new_order)))
    }

  
}


plot_stacked_histogram &lt;- function(df, genename_col, combined_pip_col, attributable_cols) {
  
  df[[genename_col]] &lt;- factor(df[[genename_col]], levels = df[[genename_col]])
  
  long_df_attributable &lt;- df %&gt;%
    select(all_of(c(genename_col, combined_pip_col, attributable_cols))) %&gt;%
    pivot_longer(
      cols = all_of(attributable_cols),
      names_to = &quot;Tissue&quot;,
      values_to = &quot;Proportion&quot;
    ) %&gt;%
    mutate(Tissue = gsub(&quot;_Attributable_PIP \\(%\\)&quot;, &quot;&quot;, Tissue))  # Modify Tissue names
  
  
  # Calculate the absolute values for each segment
  long_df_attributable &lt;- long_df_attributable %&gt;%
    mutate(Absolute_Value = Proportion / 100 * long_df_attributable[[combined_pip_col]])
  
  # Create the plot
  p &lt;- ggplot(long_df_attributable, aes_string(x = genename_col, y = &quot;Absolute_Value&quot;, fill = &quot;Tissue&quot;)) +
    geom_bar(stat = &quot;identity&quot;) +
    theme_minimal() +
    labs(title = &quot;Stacked Histogram of Combined PIP with Tissue Contributions&quot;,
         x = &quot;Gene Name&quot;,
         y = &quot;Combined PIP&quot;,
         fill = &quot;Tissue&quot;) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1))
  
  return(p)
}</code></pre>
</div>
</div>
<div id="lb" class="section level1">
<h1>LB</h1>
<pre class="r"><code>outputdir &lt;- &quot;/project/xinhe/xsun/ctwas/2.TIBETN/output/multi/LB/&quot;
outname &lt;- &quot;LB_ctwas&quot;
gwas_n &lt;- 966
weight &lt;- &quot;Lung,Whole_Blood,Heart_Left_Ventricle,Artery_Coronary,Kidney_Cortex&quot;

load(paste0(outputdir,outname,&quot;_z_snp.Rd&quot;))
load(paste0(outputdir,outname,&quot;_z_gene.Rd&quot;))

ctwas_parameters &lt;- ctwas:::ctwas_summarize_parameters(outputdir = outputdir,
                                               outname = outname,
                                               gwas_n = gwas_n,
                                               thin = thin)

ctwas_parameters$convergence_plot</code></pre>
<p><img src="figure/multi_tissue.Rmd/unnamed-chunk-2-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ctwas_res &lt;- process_data(weight = weight,outputdir = outputdir,outname = outname,z_snp = z_snp,z_gene = z_gene)
para &lt;- process_ctwas_results(ctwas_parameters = ctwas_parameters,ctwas_res = ctwas_res)
DT::datatable(para,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Parameters&#39;))</code></pre>
<div id="htmlwidget-f56a676a398ed156339f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-f56a676a398ed156339f">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Parameters<\/caption>","data":[["1","2","3","4","5","6"],["Artery_Coronary","Heart_Left_Ventricle","Kidney_Cortex","Lung","SNP","Whole_Blood"],[0.0523082143517459,0.0516539056872178,0.0262257259745829,0.11836509753816,null,0.0593344361453628],["4626","4776","3357","5644","2694570","4772"],["0.00290399093005751","0.00272958629389969","0.00238804092995241","0.00331846265332097","0.000345166495244458","0.00284920546987012"],["0.428215071381691","0.304870167304117","0.293357858900644","0.603266209000178","0.690910230198192","0.375587738462936"],["8.41330479657597","7.90802795609265","6.91851892594941","9.61409261629155",null,"8.25458295960106"],["0.00595505403047394","0.00411432836244354","0.00243452202641341","0.0116964969715901","0.665215867434734","0.00528638128423098"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>type<\/th>\n      <th>max_susie_pip<\/th>\n      <th>group_size<\/th>\n      <th>group_prior<\/th>\n      <th>group_prior_var<\/th>\n      <th>enrichment<\/th>\n      <th>group_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>df_gene &lt;- process_gene_data(ctwas_res = ctwas_res)

df_gene &lt;- attributable_pip(df_gene,combined_pip_col = &quot;combined_pip&quot;)

DT::datatable(df_gene,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;top20 Genes sorted bu susiepip, 0 means pip=0; NA means no such gene in this tissue&#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-6a5b628072b4a50f9326" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-6a5b628072b4a50f9326">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">top20 Genes sorted bu susiepip, 0 means pip=0; NA means no such gene in this tissue<\/caption>","data":[["344","3908","9260","7731","9501","7920","4116","9181","3564","959","4637","9229","1555","9405","1038","919","4114","5337","1858","6989"],["ALDH3B1","IPO8","TSC1","SCCPDH","URGCP","SH3BP5L","KIAA1841","TRDN","HELB","C14orf2","LPIN1","TRIOBP","CFAP57","UBE2D4","C1orf52","BTRC","KIAA1683","NADSYN1","CPSF1","RP11-111M22.4"],[0.223668880966501,0.176208551637002,0.152397370699996,0.147677378143024,0.147243126252414,0.146191087300554,0.14499302203413,0.144396795074524,0.144376920459519,0.136837306425571,0.136615138272083,0.13542542437112,0.134770167331131,0.132696982837296,0.131928199173034,0.13158842687087,0.125386108592037,0.124666965330789,0.12305576477943,0.122024403663216],[0.11836509753816,0.0599816161797577,0.0429569241231044,0.0673572323579462,0.0393100150464855,0.0626196148022263,0.0414561207466718,0.0637895865511677,0.0414533533318704,0.0530954779661844,0.0456242368220131,0.0356004119240689,0.0546886852047106,0.0473830998982008,0.0634441117768687,0.0606162912481526,0.0377156592879896,0.0295475298270845,0.0312974563482522,0.0330433654805526],[52.9197879591877,34.0401278045363,28.1874443934258,45.6110700263866,26.6973515484164,42.8340851405576,28.5918040503449,44.1765944446658,28.7118974417335,38.8019023124109,33.3961795150019,26.2878348651207,40.5792218617198,35.7077447316936,48.0898793241744,46.0650626271538,30.0796154466387,23.7011703530952,25.4335555951815,27.0793091288127],[0.0593344361453628,0.0234680330710046,0.02882044732314,0.0123837801530967,0.0322871265452124,null,0.0279301283897161,0.0379669328209025,0.028112086516238,0.0184577240211594,0.0307667360674909,0.0238024786369113,null,0.0269242722610357,0.0140537960179663,null,0.0219749689957716,0.0254670380588811,0.0234688552672789,0.0238285129268574],[26.5278012251733,13.3183281134674,18.9113809449343,8.38569881779941,21.9277648926468,null,19.2630845249516,26.2934733428865,19.4713160709922,13.4888098160563,22.5207370549343,17.5760783083706,null,20.2900410283243,10.6526096058763,null,17.5258401768177,20.4280564552985,19.0717235469182,19.5276618541184],[null,0.0395107242927024,0.0371002923654215,null,0.0291432983505221,0.0486651440485557,0.0303626656127294,0.042640275702454,0.0305083437220611,0.0369722958537452,0.0334398245918534,0.0307465571383129,0.0384031023540494,0.0355684911589973,0.0426885708618908,0.0407384037775219,0.0260487848076957,0.0275061821622511,0.0281453766316886,0.0254475275640137],[null,22.4227053259574,24.3444438673787,null,19.7926375867372,33.2887215952536,20.9407771400077,29.5299322124477,21.1310392443334,27.0191637204254,24.4773932192305,22.7036815879232,28.4952546357629,26.804295318914,32.3574270925212,30.9589564570897,20.7748570397454,22.0637296249785,22.8720504741386,20.8544576331208],[0.0459693472829778,0.028664897041736,0.0219566727972407,0.0516539056872178,0.0235818834881065,0.0349063284497722,0.0243079044157443,null,0.0245276894033925,0.0283118085844819,0.0267843407907261,0.0241900810976102,0.0296218525671969,null,null,0.0302337318451953,0.0235619863281433,0.022566224850923,0.0222837065048255,0.0212884670069833],[20.552410815639,16.2675969897233,14.4075141824224,34.9775343635852,16.0156090734456,23.8771932641887,16.764878802252,null,16.9886497961907,20.6901241511074,19.6056902108332,17.8622893078922,21.9795323800526,null,null,22.9759809157565,18.7915444483613,18.1012065153316,18.1086246099625,17.4460733819596],[null,0.0245832810518012,0.0215630340910894,0.0162824599447629,0.0229208028220872,null,0.020936202869268,null,0.0197754474859575,null,null,0.0210858955742167,0.0120565272051738,0.0228211195190617,0.0117417205163087,null,0.0160847091724371,0.0195799904316496,0.0178603700273851,0.0184165306848093],[null,13.9512417663156,14.1492166118388,11.0256967922288,15.566636898754,null,14.4394554824437,null,13.6970974467503,null,null,15.5701159306933,8.94599112246472,17.1979189210681,8.90008397742812,null,12.8281428884368,15.7058370512961,14.5140457737991,15.0924980019885]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genename<\/th>\n      <th>combined_pip<\/th>\n      <th>Lung<\/th>\n      <th>Lung_Attributable_PIP (%)<\/th>\n      <th>Whole_Blood<\/th>\n      <th>Whole_Blood_Attributable_PIP (%)<\/th>\n      <th>Artery_Coronary<\/th>\n      <th>Artery_Coronary_Attributable_PIP (%)<\/th>\n      <th>Heart_Left_Ventricle<\/th>\n      <th>Heart_Left_Ventricle_Attributable_PIP (%)<\/th>\n      <th>Kidney_Cortex<\/th>\n      <th>Kidney_Cortex_Attributable_PIP (%)<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7,8,9,10,11,12]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>attributable_columns &lt;- colnames(df_gene)[seq(4,12,by=2)]

plot_stacked_histogram(df_gene, &quot;genename&quot;, &quot;combined_pip&quot;, attributable_columns)</code></pre>
<p><img src="figure/multi_tissue.Rmd/unnamed-chunk-2-4.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>table_show &lt;- ctwas_res[ctwas_res$genename %in% df_gene$genename,]
table_show &lt;- rbind(table_show,ctwas_res[ctwas_res$type!=&quot;SNP&quot; &amp; ctwas_res$susie_pip &gt; 0.8,])
table_show &lt;- table_show[!duplicated(table_show$id),]
DT::datatable(table_show,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Detailed table for genes showed above&#39;),options = list(pageLength = 5))</code></pre>
<div id="htmlwidget-3c53af4867f6c499b724" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-3c53af4867f6c499b724">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Detailed table for genes showed above<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86"],[11,1,6,1,1,10,12,11,1,14,1,1,7,11,2,9,1,6,2,12,10,12,7,1,6,19,9,14,22,7,1,2,11,7,8,2,22,12,2,10,1,11,7,9,12,14,8,12,2,11,7,2,19,11,11,12,12,2,22,11,22,7,19,8,12,7,7,11,8,19,9,9,11,22,2,12,11,14,11,8,1,19,1,1,1,1],["ENSG00000006534.15|mashr_Lung","ENSG00000143653.9|mashr_Lung","ENSG00000186439.12|mashr_Lung","ENSG00000162642.13|mashr_Lung","ENSG00000175137.10|mashr_Lung","ENSG00000166167.17|mashr_Lung","ENSG00000133704.9|mashr_Lung","ENSG00000006534.15|mashr_Whole_Blood","ENSG00000243710.7|mashr_Lung","ENSG00000156411.9|mashr_Lung","ENSG00000143653.9|mashr_Heart_Left_Ventricle","ENSG00000175137.10|mashr_Artery_Coronary","ENSG00000078967.12|mashr_Lung","ENSG00000006534.15|mashr_Heart_Left_Ventricle","ENSG00000134324.11|mashr_Lung","ENSG00000165699.13|mashr_Lung","ENSG00000162642.13|mashr_Artery_Coronary","ENSG00000186439.12|mashr_Artery_Coronary","ENSG00000162929.13|mashr_Lung","ENSG00000127311.9|mashr_Lung","ENSG00000166167.17|mashr_Artery_Coronary","ENSG00000133704.9|mashr_Artery_Coronary","ENSG00000106608.16|mashr_Lung","ENSG00000243710.7|mashr_Artery_Coronary","ENSG00000186439.12|mashr_Whole_Blood","ENSG00000130518.16|mashr_Lung","ENSG00000165699.13|mashr_Artery_Coronary","ENSG00000156411.9|mashr_Artery_Coronary","ENSG00000100106.19|mashr_Lung","ENSG00000078967.12|mashr_Artery_Coronary","ENSG00000175137.10|mashr_Heart_Left_Ventricle","ENSG00000134324.11|mashr_Artery_Coronary","ENSG00000272301.1|mashr_Lung","ENSG00000106608.16|mashr_Whole_Blood","ENSG00000071894.16|mashr_Lung","ENSG00000134324.11|mashr_Whole_Blood","ENSG00000100106.19|mashr_Artery_Coronary","ENSG00000127311.9|mashr_Artery_Coronary","ENSG00000162929.13|mashr_Artery_Coronary","ENSG00000166167.17|mashr_Heart_Left_Ventricle","ENSG00000243710.7|mashr_Heart_Left_Ventricle","ENSG00000172890.11|mashr_Lung","ENSG00000106608.16|mashr_Artery_Coronary","ENSG00000165699.13|mashr_Whole_Blood","ENSG00000133704.9|mashr_Heart_Left_Ventricle","ENSG00000156411.9|mashr_Heart_Left_Ventricle","ENSG00000071894.16|mashr_Artery_Coronary","ENSG00000127311.9|mashr_Whole_Blood","ENSG00000162929.13|mashr_Whole_Blood","ENSG00000172890.11|mashr_Artery_Coronary","ENSG00000078967.12|mashr_Whole_Blood","ENSG00000134324.11|mashr_Heart_Left_Ventricle","ENSG00000130518.16|mashr_Artery_Coronary","ENSG00000172890.11|mashr_Whole_Blood","ENSG00000272301.1|mashr_Artery_Coronary","ENSG00000133704.9|mashr_Kidney_Cortex","ENSG00000127311.9|mashr_Heart_Left_Ventricle","ENSG00000162929.13|mashr_Heart_Left_Ventricle","ENSG00000100106.19|mashr_Heart_Left_Ventricle","ENSG00000272301.1|mashr_Whole_Blood","ENSG00000100106.19|mashr_Whole_Blood","ENSG00000106608.16|mashr_Heart_Left_Ventricle","ENSG00000130518.16|mashr_Heart_Left_Ventricle","ENSG00000071894.16|mashr_Whole_Blood","ENSG00000133704.9|mashr_Whole_Blood","ENSG00000106608.16|mashr_Kidney_Cortex","ENSG00000078967.12|mashr_Kidney_Cortex","ENSG00000172890.11|mashr_Heart_Left_Ventricle","ENSG00000071894.16|mashr_Heart_Left_Ventricle","ENSG00000130518.16|mashr_Whole_Blood","ENSG00000165699.13|mashr_Heart_Left_Ventricle","ENSG00000165699.13|mashr_Kidney_Cortex","ENSG00000272301.1|mashr_Heart_Left_Ventricle","ENSG00000100106.19|mashr_Kidney_Cortex","ENSG00000162929.13|mashr_Kidney_Cortex","ENSG00000127311.9|mashr_Kidney_Cortex","ENSG00000172890.11|mashr_Kidney_Cortex","ENSG00000156411.9|mashr_Whole_Blood","ENSG00000272301.1|mashr_Kidney_Cortex","ENSG00000071894.16|mashr_Kidney_Cortex","ENSG00000143653.9|mashr_Kidney_Cortex","ENSG00000130518.16|mashr_Kidney_Cortex","ENSG00000162642.13|mashr_Whole_Blood","ENSG00000143653.9|mashr_Whole_Blood","ENSG00000243710.7|mashr_Kidney_Cortex","ENSG00000162642.13|mashr_Kidney_Cortex"],[68011539,246723566,123637348,85259743,248822510,101356322,30695741,68011539,43232297,103928437,246723418,248827033,43926148,68011539,11784824,132944194,85259743,123637348,61065981,66302475,101356322,30695741,43926148,43232297,123637348,18274051,132924617,103928437,37718496,43926562,248827033,11784824,76389880,43926562,144399893,11784824,37721936,66302475,61065981,101356322,43232297,71453498,43926148,132944194,30699514,103928437,144404103,66302475,61065981,71453075,43926148,11784824,18274051,71453498,76389880,30695741,66302926,61065981,37696844,76389880,37687070,43926148,18274051,144397708,30695741,43926148,43926148,71453498,144399893,18274051,132944033,132944194,76389880,37721936,61065981,66301720,71453075,103928103,76389880,144385418,246722724,18274051,85249770,246721765,43185070,85260420],["Lung","Lung","Lung","Lung","Lung","Lung","Lung","Whole_Blood","Lung","Lung","Heart_Left_Ventricle","Artery_Coronary","Lung","Heart_Left_Ventricle","Lung","Lung","Artery_Coronary","Artery_Coronary","Lung","Lung","Artery_Coronary","Artery_Coronary","Lung","Artery_Coronary","Whole_Blood","Lung","Artery_Coronary","Artery_Coronary","Lung","Artery_Coronary","Heart_Left_Ventricle","Artery_Coronary","Lung","Whole_Blood","Lung","Whole_Blood","Artery_Coronary","Artery_Coronary","Artery_Coronary","Heart_Left_Ventricle","Heart_Left_Ventricle","Lung","Artery_Coronary","Whole_Blood","Heart_Left_Ventricle","Heart_Left_Ventricle","Artery_Coronary","Whole_Blood","Whole_Blood","Artery_Coronary","Whole_Blood","Heart_Left_Ventricle","Artery_Coronary","Whole_Blood","Artery_Coronary","Kidney_Cortex","Heart_Left_Ventricle","Heart_Left_Ventricle","Heart_Left_Ventricle","Whole_Blood","Whole_Blood","Heart_Left_Ventricle","Heart_Left_Ventricle","Whole_Blood","Whole_Blood","Kidney_Cortex","Kidney_Cortex","Heart_Left_Ventricle","Heart_Left_Ventricle","Whole_Blood","Heart_Left_Ventricle","Kidney_Cortex","Heart_Left_Ventricle","Kidney_Cortex","Kidney_Cortex","Kidney_Cortex","Kidney_Cortex","Whole_Blood","Kidney_Cortex","Kidney_Cortex","Kidney_Cortex","Kidney_Cortex","Whole_Blood","Whole_Blood","Kidney_Cortex","Kidney_Cortex"],[11,1,6,1,1,10,12,11,1,14,1,1,7,11,2,9,1,6,2,12,10,12,7,1,6,19,9,14,22,7,1,2,11,7,8,2,22,12,2,10,1,11,7,9,12,14,8,12,2,11,7,2,19,11,11,12,12,2,22,11,22,7,19,8,12,7,7,11,8,19,9,9,11,22,2,12,11,14,11,8,1,19,1,1,1,1],[37,130,83,52,131,65,21,37,26,54,130,131,32,37,7,70,52,83,40,40,65,21,32,26,83,14,70,54,15,32,131,7,42,32,94,7,15,40,40,65,26,40,32,70,21,54,94,40,40,40,32,7,14,40,42,21,40,40,15,42,15,32,14,94,21,32,32,40,94,14,70,70,42,15,40,40,40,54,42,94,130,14,52,130,26,52],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0.11836509753816,0.0673572323579462,0.0637895865511677,0.0634441117768687,0.0626196148022263,0.0606162912481526,0.0599816161797577,0.0593344361453628,0.0546886852047106,0.0530954779661844,0.0516539056872178,0.0486651440485557,0.0473830998982008,0.0459693472829778,0.0456242368220131,0.0429569241231044,0.0426885708618908,0.042640275702454,0.0414561207466718,0.0414533533318704,0.0407384037775219,0.0395107242927024,0.0393100150464855,0.0384031023540494,0.0379669328209025,0.0377156592879896,0.0371002923654215,0.0369722958537452,0.0356004119240689,0.0355684911589973,0.0349063284497722,0.0334398245918534,0.0330433654805526,0.0322871265452124,0.0312974563482522,0.0307667360674909,0.0307465571383129,0.0305083437220611,0.0303626656127294,0.0302337318451953,0.0296218525671969,0.0295475298270845,0.0291432983505221,0.02882044732314,0.028664897041736,0.0283118085844819,0.0281453766316886,0.028112086516238,0.0279301283897161,0.0275061821622511,0.0269242722610357,0.0267843407907261,0.0260487848076957,0.0254670380588811,0.0254475275640137,0.0245832810518012,0.0245276894033925,0.0243079044157443,0.0241900810976102,0.0238285129268574,0.0238024786369113,0.0235818834881065,0.0235619863281433,0.0234688552672789,0.0234680330710046,0.0229208028220872,0.0228211195190617,0.022566224850923,0.0222837065048255,0.0219749689957716,0.0219566727972407,0.0215630340910894,0.0212884670069833,0.0210858955742167,0.020936202869268,0.0197754474859575,0.0195799904316496,0.0184577240211594,0.0184165306848093,0.0178603700273851,0.0162824599447629,0.0160847091724371,0.0140537960179663,0.0123837801530967,0.0120565272051738,0.0117417205163087],[10.6602628144169,8.05422400084898,8.10448587371618,7.97987988584558,8.0426767220943,7.99432955927313,8.37041643787865,5.98771673923008,7.15362035723295,7.30786139668783,4.58137130216757,5.8619158154374,6.81617386676892,4.55296568971676,6.32835181830794,6.39416936012272,5.37129252973748,5.45040935750175,6.33327681890398,6.24226777023474,5.38046714761382,5.61925824672165,6.10075821349547,4.84667060802273,4.64197275790553,5.74545001653334,4.8750265725584,4.9446037969591,5.78888193055564,4.78343911588428,3.81745028491221,4.32267790133827,5.37621302660991,4.08885703617483,5.22723435932416,3.70674962593852,4.39397428681846,4.26802004001837,4.32580496152658,3.52509660183072,3.20095532822189,4.91536308304098,4.17817052781703,3.74140584313506,3.65786662962913,3.26042403256143,4.06975077015053,3.66142208278023,3.70934288429553,3.92624262560847,3.58691024598842,2.88276730954993,3.68080615033851,3.36362194999276,3.7181307035849,3.49203229754145,2.8495540108233,2.88466617774579,2.89659990740228,3.20540108181951,3.29545117670386,2.7950170140255,2.67068131363488,3.20554396831607,3.67766163258474,2.94146208259345,2.93148401987986,2.63151805426014,2.65935953441923,2.88381618659756,2.59801485983744,2.77391967296185,2.51566358350005,2.79080612676951,2.75179303551651,2.56664139585909,2.52312476271657,2.58614720885897,2.40402296187025,2.36890575951001,2.08920171473599,2.0130527226365,1.82659065595538,1.36536154219011,1.34555741595122,1.43815863292467],["ALDH3B1","SCCPDH","TRDN","C1orf52","SH3BP5L","BTRC","IPO8","ALDH3B1","CFAP57","C14orf2","SCCPDH","SH3BP5L","UBE2D4","ALDH3B1","LPIN1","TSC1","C1orf52","TRDN","KIAA1841","HELB","BTRC","IPO8","URGCP","CFAP57","TRDN","KIAA1683","TSC1","C14orf2","TRIOBP","UBE2D4","SH3BP5L","LPIN1","RP11-111M22.4","URGCP","CPSF1","LPIN1","TRIOBP","HELB","KIAA1841","BTRC","CFAP57","NADSYN1","URGCP","TSC1","IPO8","C14orf2","CPSF1","HELB","KIAA1841","NADSYN1","UBE2D4","LPIN1","KIAA1683","NADSYN1","RP11-111M22.4","IPO8","HELB","KIAA1841","TRIOBP","RP11-111M22.4","TRIOBP","URGCP","KIAA1683","CPSF1","IPO8","URGCP","UBE2D4","NADSYN1","CPSF1","KIAA1683","TSC1","TSC1","RP11-111M22.4","TRIOBP","KIAA1841","HELB","NADSYN1","C14orf2","RP11-111M22.4","CPSF1","SCCPDH","KIAA1683","C1orf52","SCCPDH","CFAP57","C1orf52"],[-4.07180968742576,-3.2799059986616,-3.18367136565155,3.18510489714066,-3.48260177489866,-3.32796502128879,3.55999934768981,-4.07180968742576,2.8388619066705,-2.94096302060543,-3.90744786345332,-3.65421491528208,-2.92460105051897,-4.07180968742576,-2.58297443562086,2.84090229569934,3.18510489714066,-3.18367136565155,2.68939221093762,-2.69322840293569,-3.32796502128879,3.55999934768981,2.72453408383936,2.8388619066705,-3.18367136565155,2.47338674926881,3.0714865619582,-2.94096302060543,2.6674411811575,-2.97747534555801,-3.65421491528208,-2.58297443562086,2.30146009719113,2.97747534555801,2.38879616396977,-2.58297443562086,2.85528133084157,-2.69322840293569,2.68939221093762,-3.32796502128879,2.83886190667049,2.40626499150569,2.72453408383936,2.84090229569934,3.55285451649669,-2.94096302060543,2.6089374628699,-2.6932284029357,2.68939221093762,2.69613474607326,-2.72453408383936,-2.58297443562086,2.33982608631408,2.68782243212838,2.30146009719113,3.55999934768981,-2.69322840293569,2.68939221093762,2.83223004630607,2.30146009719113,2.59183305788549,2.72453408383936,2.48250549151667,2.42657800474019,2.95332935147188,2.9344815064486,-2.92755676072945,2.68782243212838,2.55510290147549,2.13918491671963,2.53927700205689,2.84090229569934,2.30146009719113,2.85528133084157,2.68939221093762,-2.56955659060618,2.69613474607326,-1.90520241436811,2.30146009719113,2.40420208752239,-2.15141753896545,1.96928314335618,1.19408981857077,0.0115293623254738,0.970724684856966,-1.150499010211]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chrom<\/th>\n      <th>id<\/th>\n      <th>pos<\/th>\n      <th>type<\/th>\n      <th>region_tag1<\/th>\n      <th>region_tag2<\/th>\n      <th>cs_index<\/th>\n      <th>susie_pip<\/th>\n      <th>mu2<\/th>\n      <th>genename<\/th>\n      <th>z<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,3,5,6,7,8,9,11]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="mc" class="section level1">
<h1>MC</h1>
<pre class="r"><code>outputdir &lt;- &quot;/project/xinhe/xsun/ctwas/2.TIBETN/output/multi/MC/&quot;
outname &lt;- &quot;MC_ctwas&quot;
gwas_n &lt;- 966
weight &lt;- &quot;Liver,Artery_Tibial,Heart_Atrial_Appendage,Pancreas,Lung&quot;

load(paste0(outputdir,outname,&quot;_z_snp.Rd&quot;))
load(paste0(outputdir,outname,&quot;_z_gene.Rd&quot;))

ctwas_parameters &lt;- ctwas:::ctwas_summarize_parameters(outputdir = outputdir,
                                               outname = outname,
                                               gwas_n = gwas_n,
                                               thin = thin)

ctwas_parameters$convergence_plot</code></pre>
<p><img src="figure/multi_tissue.Rmd/unnamed-chunk-3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ctwas_res &lt;- process_data(weight = weight,outputdir = outputdir,outname = outname,z_snp = z_snp,z_gene = z_gene)
para &lt;- process_ctwas_results(ctwas_parameters = ctwas_parameters,ctwas_res = ctwas_res)
DT::datatable(para,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Parameters&#39;))</code></pre>
<div id="htmlwidget-14469a047db901fc1b8f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-14469a047db901fc1b8f">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Parameters<\/caption>","data":[["1","2","3","4","5","6"],["Artery_Tibial","Heart_Atrial_Appendage","Liver","Lung","Pancreas","SNP"],[0.0882468972284088,0.0568361729907803,0.0848895645506954,0.0651170155780841,0.0436042389944755,null],["5672","5178","4261","5644","4829","2694570"],["0.00337344935481699","0.00320165373470793","0.00472800012421472","0.00285157696963644","0.00297894926664042","0.000358968290384463"],["0.603000385571891","0.449708343089998","0.625487091261182","0.394680565744463","0.379382793660556","0.795934034069635"],["9.39762493005705","8.91904332630298","13.1710801507034","7.94381299412919","8.29864182000561",null],["0.0119440298510831","0.00771774144041168","0.01304458413613","0.00657568073881584","0.00564964053469866","0.796976482074747"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>type<\/th>\n      <th>max_susie_pip<\/th>\n      <th>group_size<\/th>\n      <th>group_prior<\/th>\n      <th>group_prior_var<\/th>\n      <th>enrichment<\/th>\n      <th>group_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>df_gene &lt;- process_gene_data(ctwas_res = ctwas_res)

df_gene &lt;- attributable_pip(df_gene,combined_pip_col = &quot;combined_pip&quot;)

DT::datatable(df_gene,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;top20 Genes sorted bu susiepip, 0 means pip=0; NA means no such gene in this tissue&#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-2bbd72fbe74588e1d59a" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-2bbd72fbe74588e1d59a">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">top20 Genes sorted bu susiepip, 0 means pip=0; NA means no such gene in this tissue<\/caption>","data":[["9081","2945","2832","6203","1281","6654","9119","70","4884","6953","4337","1133","599","7754","6080","9912","8992","8310","7218","783"],["SYS1","FAAP24","EPSTI1","PAWR","CAP1","PP2D1","TANGO6","AC002064.5","LLPH","PYROXD2","KCTD3","C22orf39","ARHGEF19","RP11-575F12.3","ORC3","UBLCP1","STPG1","SF3B3","RHOQ","ATP7B"],[0.201246523087476,0.199894190205284,0.197064575863232,0.186724010344168,0.173290248386343,0.164290099747358,0.164178449431664,0.163970003862122,0.161074632841564,0.153014187192647,0.152699827694287,0.150630259575367,0.15055499872051,0.150405288862661,0.149460620965616,0.148560832178605,0.145441942529576,0.143799555625271,0.142686588924305,0.142223552360344],[0.0439807771882045,0.0518735709891049,0.0528401386674639,0.0133173048573975,0.0447771217730969,0.0292756404703881,0.0381316280006494,0.0759514511896212,0.0501253949609124,0.0320922040438634,null,0.0552322845693277,0.0155294816404228,0.014912069095136,0.0373210987456082,0.0309017085158411,0.0312914635404432,0.0369140053287035,0.0343249703963602,0.0271584124820149],[21.8541798951167,25.950514587659,26.8136160119089,7.13207949682056,25.8393776857358,17.8194793937113,23.2257206306001,46.3203326222319,31.1193600610077,20.9733519699444,null,36.6674562767333,10.3148230031549,9.91459090827094,24.9705230076583,20.8007104313269,21.5147453315126,25.6704585547525,24.0561994334096,19.0955801843601],[0.0789816767758402,0.0368356035455567,0.0752198846357146,0.0700682785782547,0.0644648577460221,0.0511967931205988,0.0542481487093548,null,0.0428750752122717,0.045380255030662,0.0595904222660956,null,0.0592613591403741,0.0690147877976844,0.0527324325602077,0.0435522447767523,0.0433428541750557,0.0524884419667648,0.0570613943869475,0.0458507807567479],[39.2462317182539,18.427550849641,38.170170517059,37.5250501791953,37.2005109037067,31.1624335241918,33.0421860464302,null,26.6181424448407,29.657547357702,39.0245510855447,null,39.3619339404245,45.8858782956122,35.2818235462429,29.31610178677,29.8007943384294,36.5011155552838,39.9907200929854,32.2385287076631],[0.0209228947332923,0.034206981519867,0.0341664333188809,0.0319635010506829,0.0182204946779093,0.0274961318802525,0.0150411556686714,0.0381319512330116,0.0208530364357048,0.0233094342760102,0.028936326941248,0.0121076818739048,0.0333804009778627,0.0249025233520969,0.0268807239545275,0.0231875775970525,0.0177487371706583,0.0245373592324687,0.0175646674436939,0.0233386669420094],[10.3966490512722,17.1125441338429,17.3376839389913,17.1180455002911,10.5144373948195,16.7363291656256,9.16146773266485,23.2554432730732,12.9461952312604,15.2335118093745,18.9498098185023,8.03801434588032,22.1715660466578,16.5569465943688,17.9851547389941,15.6081365841944,12.2033141623153,17.0635848808955,12.3099637997597,16.4098467199564],[0.0432276698783762,0.0418220119188979,null,0.0387684273944593,0.0274027325563642,0.026062939544373,0.0306721322616438,0.0498866014394888,0.0242617183676297,0.028234596616612,0.0345039655310392,0.0430426963249156,0.0423837569618505,0.0257516933303099,0.0157782125053184,0.0269649288748297,0.0298825403865517,0.0298597490973341,0.0204252916064673,0.0274720019916731],[21.4799586175143,20.9220747616268,null,20.7624222096568,15.8131994220881,15.8639745087818,18.6821914616817,30.4242241046949,15.0624079903966,18.4522736973823,22.5959426752714,28.5750661561991,28.1516770097628,17.121534438742,10.5567690026848,18.1507659047114,20.5460267284831,20.7648410090682,14.3147942357098,19.3160707461931],[0.0141335045117628,0.0351560222318573,0.0348381192411722,0.0326064984633734,0.0184250416329508,0.0302585947317453,0.0260853847913449,null,0.0229594078650454,0.0239976972254991,0.0296691129559038,0.0402475968072193,null,0.0158242152874335,0.0167481531999545,0.023954372414129,0.0231763472568672,null,0.0133102650908362,0.018403690187899],[7.02298071784295,17.5873156672304,17.6785295320407,17.4624026140363,10.6324745936499,18.4177834076895,15.8884341286232,null,14.2538942724946,15.6833151655968,19.4296964206816,26.7194632211873,null,10.521049763006,11.2057297044199,16.1242852929972,15.9351194392596,null,9.32832243813556,12.9399736418273]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genename<\/th>\n      <th>combined_pip<\/th>\n      <th>Artery_Tibial<\/th>\n      <th>Artery_Tibial_Attributable_PIP (%)<\/th>\n      <th>Liver<\/th>\n      <th>Liver_Attributable_PIP (%)<\/th>\n      <th>Lung<\/th>\n      <th>Lung_Attributable_PIP (%)<\/th>\n      <th>Heart_Atrial_Appendage<\/th>\n      <th>Heart_Atrial_Appendage_Attributable_PIP (%)<\/th>\n      <th>Pancreas<\/th>\n      <th>Pancreas_Attributable_PIP (%)<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7,8,9,10,11,12]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>attributable_columns &lt;- colnames(df_gene)[seq(4,12,by=2)]

plot_stacked_histogram(df_gene, &quot;genename&quot;, &quot;combined_pip&quot;, attributable_columns)</code></pre>
<p><img src="figure/multi_tissue.Rmd/unnamed-chunk-3-4.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>table_show &lt;- ctwas_res[ctwas_res$genename %in% df_gene$genename,]
table_show &lt;- rbind(table_show,ctwas_res[ctwas_res$type!=&quot;SNP&quot; &amp; ctwas_res$susie_pip &gt; 0.8,])
table_show &lt;- table_show[!duplicated(table_show$id),]
DT::datatable(table_show,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Detailed table for genes showed above&#39;),options = list(pageLength = 5))</code></pre>
<div id="htmlwidget-527d026f50f3b60341d6" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-527d026f50f3b60341d6">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Detailed table for genes showed above<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93"],[20,7,13,12,12,1,1,1,2,22,16,13,6,16,19,3,12,7,13,10,1,20,5,1,20,22,12,1,19,22,12,7,16,6,16,19,19,13,1,2,19,13,1,12,10,12,1,5,16,3,1,16,1,3,1,10,3,13,1,13,5,6,16,3,12,12,16,12,10,5,13,10,5,1,12,20,12,2,1,13,1,1,2,6,12,6,1,16,12,20,12,2,22],["ENSG00000204070.9|mashr_Liver","ENSG00000225498.1|mashr_Artery_Tibial","ENSG00000133106.14|mashr_Liver","ENSG00000177425.10|mashr_Liver","ENSG00000278266.1|mashr_Liver","ENSG00000131236.16|mashr_Liver","ENSG00000136636.12|mashr_Liver","ENSG00000142632.16|mashr_Liver","ENSG00000119729.11|mashr_Liver","ENSG00000242259.8|mashr_Artery_Tibial","ENSG00000103047.7|mashr_Liver","ENSG00000133106.14|mashr_Artery_Tibial","ENSG00000135336.14|mashr_Liver","ENSG00000189091.12|mashr_Liver","ENSG00000131944.9|mashr_Artery_Tibial","ENSG00000183977.13|mashr_Liver","ENSG00000139233.6|mashr_Artery_Tibial","ENSG00000225498.1|mashr_Heart_Atrial_Appendage","ENSG00000123191.14|mashr_Liver","ENSG00000119943.12|mashr_Liver","ENSG00000131236.16|mashr_Artery_Tibial","ENSG00000204070.9|mashr_Artery_Tibial","ENSG00000164332.7|mashr_Liver","ENSG00000001460.17|mashr_Liver","ENSG00000204070.9|mashr_Heart_Atrial_Appendage","ENSG00000242259.8|mashr_Heart_Atrial_Appendage","ENSG00000139233.6|mashr_Liver","ENSG00000142632.16|mashr_Heart_Atrial_Appendage","ENSG00000131944.9|mashr_Heart_Atrial_Appendage","ENSG00000242259.8|mashr_Pancreas","ENSG00000177425.10|mashr_Heart_Atrial_Appendage","ENSG00000225498.1|mashr_Lung","ENSG00000103047.7|mashr_Artery_Tibial","ENSG00000135336.14|mashr_Artery_Tibial","ENSG00000189091.12|mashr_Artery_Tibial","ENSG00000131944.9|mashr_Liver","ENSG00000131944.9|mashr_Pancreas","ENSG00000133106.14|mashr_Pancreas","ENSG00000136636.12|mashr_Heart_Atrial_Appendage","ENSG00000119729.11|mashr_Artery_Tibial","ENSG00000131944.9|mashr_Lung","ENSG00000133106.14|mashr_Lung","ENSG00000142632.16|mashr_Lung","ENSG00000177425.10|mashr_Pancreas","ENSG00000119943.12|mashr_Artery_Tibial","ENSG00000177425.10|mashr_Lung","ENSG00000001460.17|mashr_Artery_Tibial","ENSG00000164332.7|mashr_Artery_Tibial","ENSG00000103047.7|mashr_Heart_Atrial_Appendage","ENSG00000183977.13|mashr_Pancreas","ENSG00000001460.17|mashr_Heart_Atrial_Appendage","ENSG00000189091.12|mashr_Heart_Atrial_Appendage","ENSG00000136636.12|mashr_Pancreas","ENSG00000183977.13|mashr_Artery_Tibial","ENSG00000136636.12|mashr_Lung","ENSG00000119943.12|mashr_Heart_Atrial_Appendage","ENSG00000183977.13|mashr_Lung","ENSG00000123191.14|mashr_Heart_Atrial_Appendage","ENSG00000131236.16|mashr_Heart_Atrial_Appendage","ENSG00000123191.14|mashr_Artery_Tibial","ENSG00000164332.7|mashr_Heart_Atrial_Appendage","ENSG00000135336.14|mashr_Lung","ENSG00000103047.7|mashr_Pancreas","ENSG00000183977.13|mashr_Heart_Atrial_Appendage","ENSG00000278266.1|mashr_Heart_Atrial_Appendage","ENSG00000278266.1|mashr_Lung","ENSG00000189091.12|mashr_Lung","ENSG00000139233.6|mashr_Heart_Atrial_Appendage","ENSG00000119943.12|mashr_Pancreas","ENSG00000164332.7|mashr_Pancreas","ENSG00000123191.14|mashr_Lung","ENSG00000119943.12|mashr_Lung","ENSG00000164332.7|mashr_Lung","ENSG00000001460.17|mashr_Pancreas","ENSG00000139233.6|mashr_Pancreas","ENSG00000204070.9|mashr_Lung","ENSG00000139233.6|mashr_Lung","ENSG00000119729.11|mashr_Heart_Atrial_Appendage","ENSG00000131236.16|mashr_Pancreas","ENSG00000123191.14|mashr_Pancreas","ENSG00000131236.16|mashr_Lung","ENSG00000001460.17|mashr_Lung","ENSG00000119729.11|mashr_Lung","ENSG00000135336.14|mashr_Pancreas","ENSG00000278266.1|mashr_Pancreas","ENSG00000135336.14|mashr_Heart_Atrial_Appendage","ENSG00000142632.16|mashr_Artery_Tibial","ENSG00000103047.7|mashr_Lung","ENSG00000278266.1|mashr_Artery_Tibial","ENSG00000204070.9|mashr_Pancreas","ENSG00000177425.10|mashr_Artery_Tibial","ENSG00000119729.11|mashr_Pancreas","ENSG00000242259.8|mashr_Lung"],[45362193,90312570,42992237,79690683,127147075,40039889,215567710,16210794,46541511,19443873,68843188,42992237,87589592,70524136,32972322,20012243,66130421,90312570,52011745,98414804,40039889,45360394,159268366,24417293,45362193,19443515,66131133,16213318,32972339,19454219,79690683,90312570,68843188,87589592,70524136,32972339,32972152,42992237,215567710,46547460,32972339,42992237,16200034,79690683,98414964,79690683,24416496,159268366,68843188,19971502,24417293,70524136,215567710,19982498,215567710,98387340,20012089,52011745,40040200,52011745,159267588,87586421,68843188,20012089,127147075,127147629,70480925,66130421,98415303,159267588,52011745,98415303,159267588,24417389,66131133,45361414,66130421,46541511,40040200,52011745,40039999,24416496,46541511,87588120,127146585,87590295,16202622,68842808,127165406,45361414,79621793,46541996,19341828],["Liver","Artery_Tibial","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Artery_Tibial","Liver","Artery_Tibial","Liver","Liver","Artery_Tibial","Liver","Artery_Tibial","Heart_Atrial_Appendage","Liver","Liver","Artery_Tibial","Artery_Tibial","Liver","Liver","Heart_Atrial_Appendage","Heart_Atrial_Appendage","Liver","Heart_Atrial_Appendage","Heart_Atrial_Appendage","Pancreas","Heart_Atrial_Appendage","Lung","Artery_Tibial","Artery_Tibial","Artery_Tibial","Liver","Pancreas","Pancreas","Heart_Atrial_Appendage","Artery_Tibial","Lung","Lung","Lung","Pancreas","Artery_Tibial","Lung","Artery_Tibial","Artery_Tibial","Heart_Atrial_Appendage","Pancreas","Heart_Atrial_Appendage","Heart_Atrial_Appendage","Pancreas","Artery_Tibial","Lung","Heart_Atrial_Appendage","Lung","Heart_Atrial_Appendage","Heart_Atrial_Appendage","Artery_Tibial","Heart_Atrial_Appendage","Lung","Pancreas","Heart_Atrial_Appendage","Heart_Atrial_Appendage","Lung","Lung","Heart_Atrial_Appendage","Pancreas","Pancreas","Lung","Lung","Lung","Pancreas","Pancreas","Lung","Lung","Heart_Atrial_Appendage","Pancreas","Pancreas","Lung","Lung","Lung","Pancreas","Pancreas","Heart_Atrial_Appendage","Artery_Tibial","Lung","Artery_Tibial","Pancreas","Artery_Tibial","Pancreas","Lung"],[20,7,13,12,12,1,1,1,2,22,16,13,6,16,19,3,12,7,13,10,1,20,5,1,20,22,12,1,19,22,12,7,16,6,16,19,19,13,1,2,19,13,1,12,10,12,1,5,16,3,1,16,1,3,1,10,3,13,1,13,5,6,16,3,12,12,16,12,10,5,13,10,5,1,12,20,12,2,1,13,1,1,2,6,12,6,1,16,12,20,12,2,22],[28,55,17,49,77,25,109,11,29,3,37,17,59,37,23,14,40,55,22,62,25,28,93,17,28,3,40,11,23,3,49,55,37,59,37,23,23,17,109,29,23,17,11,49,62,49,17,93,37,14,17,37,109,14,109,62,14,22,25,22,93,59,37,14,77,77,37,40,62,93,22,62,93,17,40,28,40,29,25,22,25,17,29,59,77,59,11,37,77,28,49,29,3],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0.0789816767758402,0.0759514511896212,0.0752198846357146,0.0700682785782547,0.0690147877976844,0.0644648577460221,0.0595904222660956,0.0592613591403741,0.0570613943869475,0.0552322845693277,0.0542481487093548,0.0528401386674639,0.0527324325602077,0.0524884419667648,0.0518735709891049,0.0511967931205988,0.0501253949609124,0.0498866014394888,0.0458507807567479,0.045380255030662,0.0447771217730969,0.0439807771882045,0.0435522447767523,0.0433428541750557,0.0432276698783762,0.0430426963249156,0.0428750752122717,0.0423837569618505,0.0418220119188979,0.0402475968072193,0.0387684273944593,0.0381319512330116,0.0381316280006494,0.0373210987456082,0.0369140053287035,0.0368356035455567,0.0351560222318573,0.0348381192411722,0.0345039655310392,0.0343249703963602,0.034206981519867,0.0341664333188809,0.0333804009778627,0.0326064984633734,0.0320922040438634,0.0319635010506829,0.0312914635404432,0.0309017085158411,0.0306721322616438,0.0302585947317453,0.0298825403865517,0.0298597490973341,0.0296691129559038,0.0292756404703881,0.028936326941248,0.028234596616612,0.0274961318802525,0.0274720019916731,0.0274027325563642,0.0271584124820149,0.0269649288748297,0.0268807239545275,0.0260853847913449,0.026062939544373,0.0257516933303099,0.0249025233520969,0.0245373592324687,0.0242617183676297,0.0239976972254991,0.023954372414129,0.0233386669420094,0.0233094342760102,0.0231875775970525,0.0231763472568672,0.0229594078650454,0.0209228947332923,0.0208530364357048,0.0204252916064673,0.0184250416329508,0.018403690187899,0.0182204946779093,0.0177487371706583,0.0175646674436939,0.0167481531999545,0.0158242152874335,0.0157782125053184,0.0155294816404228,0.0150411556686714,0.014912069095136,0.0141335045117628,0.0133173048573975,0.0133102650908362,0.0121076818739048],[7.58457090515671,11.0892201457914,7.41126571289301,7.32978695751411,7.3440176296831,6.88340627959197,6.37517819056687,6.42770026172369,6.33141577925499,7.29570481106939,6.83426213433485,7.12474850982509,5.60713748155888,6.70454454590344,6.91233466399634,6.08196194823613,7.05867341267605,7.81306546156644,5.639720498196,5.58746482254446,6.57090210177603,6.40035507169512,4.97198954306032,5.29833063125533,5.22970417376414,5.23333387477924,5.33566526324818,5.15184902168224,5.02534913765505,4.58209439892459,5.06412602498238,6.62637919049311,6.57333578992065,5.40063571041711,6.44937134513308,4.45741949690118,4.1004296743358,4.17850650671754,4.44374803843953,5.4856269670017,4.27970027223993,4.38270715732932,4.28650241808315,4.13688091831327,5.38184783982801,4.33863982919383,5.1788852093768,4.79365740850718,4.74209599264879,3.92161163220115,4.13226603108624,4.65779555553085,3.64919365019947,5.05260501196228,3.82234502843921,4.04217473697596,3.90013946976226,3.96579122704383,3.87940860434188,4.75786339260323,3.53156513392651,3.40072704625527,3.8837288882186,3.8081721441779,3.83830878855718,3.678623011542,3.96036729543288,3.54310559251136,3.26187935388013,2.93211471063051,3.42457685796623,3.41231103632575,3.06320253603423,3.08717672358907,3.11813083447209,2.96441758455309,3.07280680777651,2.90201430082943,2.4677619880736,2.53409050066362,2.64654823552273,2.55438686031155,2.49006256374458,1.85720501098056,2.18264859564248,1.76187440803741,2.42159082780077,2.56447347492769,2.57524394264832,1.65922778325014,2.07487016559066,1.51717075433226,1.42199798180381],["SYS1","AC002064.5","EPSTI1","PAWR","RP11-575F12.3","CAP1","KCTD3","ARHGEF19","RHOQ","C22orf39","TANGO6","EPSTI1","ORC3","SF3B3","FAAP24","PP2D1","LLPH","AC002064.5","ATP7B","PYROXD2","CAP1","SYS1","UBLCP1","STPG1","SYS1","C22orf39","LLPH","ARHGEF19","FAAP24","C22orf39","PAWR","AC002064.5","TANGO6","ORC3","SF3B3","FAAP24","FAAP24","EPSTI1","KCTD3","RHOQ","FAAP24","EPSTI1","ARHGEF19","PAWR","PYROXD2","PAWR","STPG1","UBLCP1","TANGO6","PP2D1","STPG1","SF3B3","KCTD3","PP2D1","KCTD3","PYROXD2","PP2D1","ATP7B","CAP1","ATP7B","UBLCP1","ORC3","TANGO6","PP2D1","RP11-575F12.3","RP11-575F12.3","SF3B3","LLPH","PYROXD2","UBLCP1","ATP7B","PYROXD2","UBLCP1","STPG1","LLPH","SYS1","LLPH","RHOQ","CAP1","ATP7B","CAP1","STPG1","RHOQ","ORC3","RP11-575F12.3","ORC3","ARHGEF19","TANGO6","RP11-575F12.3","SYS1","PAWR","RHOQ","C22orf39"],[3.1536556615109,4.82721785522274,2.92577428293139,3.27251618441354,-3.23531224341598,-3.1960344613078,2.74739467338896,-2.65880512383365,-2.57310134194843,-3.00475481165443,-3.04368156856893,-2.92577428293139,-2.59005379249351,2.99229392817754,2.93625112183979,2.62386978140543,2.93426183548587,4.82721785522274,-2.44975239353755,2.61196106780215,2.99482864575114,-2.88699060861376,2.16844821149455,-2.37258146528037,3.1536556615109,-3.0047548116545,-2.33702418702747,-2.92003043463918,2.95811830793809,-3.13551524297172,-3.27251618441354,4.82721785522274,-3.04368156856893,-2.59005379249351,2.99229392817754,2.04478126092839,2.95540955488093,2.92577428293139,2.74739467338896,-2.39017335320457,2.94481113606234,-2.92577428293139,-2.86955408820035,3.27251618441354,2.61196106780073,3.27251618441354,2.39870937680005,2.16844821149455,-3.04368156856893,2.86805104181134,2.56395344639248,2.99229392817754,2.74739467338896,2.33752380510584,2.74739467338896,2.65764038178548,2.75511646158078,-2.44975239353755,2.79719199115759,-2.21032384707232,2.16844821149328,-2.58608835278568,-3.04368156856893,2.39341323255635,2.60638399172589,-2.63300661368261,2.97703243280456,2.19707343415622,2.62086502394378,2.16844821149328,-2.44975239353755,2.62086502394378,2.16844821149328,2.36493055193442,-2.33702418702747,2.32779781218276,2.19707343415622,-1.77661480880287,2.2451346218337,-1.92957404980505,2.3060587112301,1.83498126976476,-1.73779191634455,-1.42609470294929,1.79959638956168,0.761808818102611,-0.975890643893314,-2.14706778189912,-1.06788697675396,1.14913112980126,0.518859446016163,-0.661339226229186,0.031031707328323]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chrom<\/th>\n      <th>id<\/th>\n      <th>pos<\/th>\n      <th>type<\/th>\n      <th>region_tag1<\/th>\n      <th>region_tag2<\/th>\n      <th>cs_index<\/th>\n      <th>susie_pip<\/th>\n      <th>mu2<\/th>\n      <th>genename<\/th>\n      <th>z<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,3,5,6,7,8,9,11]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="oxhb" class="section level1">
<h1>OxHb</h1>
<pre class="r"><code>outputdir &lt;- &quot;/project/xinhe/xsun/ctwas/2.TIBETN/output/multi/OxHb/&quot;
outname &lt;- &quot;OxHb_ctwas&quot;
gwas_n &lt;- 966
weight &lt;- &quot;Heart_Left_Ventricle,Kidney_Cortex,Liver,Artery_Coronary,Lung&quot;

load(paste0(outputdir,outname,&quot;_z_snp.Rd&quot;))
load(paste0(outputdir,outname,&quot;_z_gene.Rd&quot;))

ctwas_parameters &lt;- ctwas:::ctwas_summarize_parameters(outputdir = outputdir,
                                               outname = outname,
                                               gwas_n = gwas_n,
                                               thin = thin)

ctwas_parameters$convergence_plot</code></pre>
<p><img src="figure/multi_tissue.Rmd/unnamed-chunk-4-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ctwas_res &lt;- process_data(weight = weight,outputdir = outputdir,outname = outname,z_snp = z_snp,z_gene = z_gene)
para &lt;- process_ctwas_results(ctwas_parameters = ctwas_parameters,ctwas_res = ctwas_res)
DT::datatable(para,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Parameters&#39;))</code></pre>
<div id="htmlwidget-70591edb04dd70f97856" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-70591edb04dd70f97856">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Parameters<\/caption>","data":[["1","2","3","4","5","6"],["Artery_Coronary","Heart_Left_Ventricle","Kidney_Cortex","Liver","Lung","SNP"],[0.106329955878981,0.373307992489503,0.101985882066122,0.0808753818379009,0.0550308327257063,null],["4625","4778","3353","4258","5642","2694310"],["0.00388574145578627","0.00547781648228912","0.002806317315398","0.00294907477117959","0.00269119485296925","0.000346132351944032"],["0.529227762406249","2.8790586952266","0.93884085529064","0.634901428860654","0.411809665948891","0.654504673624024"],["11.2261723989746","15.8257858634805","8.10764235020647","8.52007838798161","7.77504569525014",null],["0.00984580272639671","0.0780058217621948","0.00914503102885426","0.00825316673390613","0.00647288118147929","0.631866574617364"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>type<\/th>\n      <th>max_susie_pip<\/th>\n      <th>group_size<\/th>\n      <th>group_prior<\/th>\n      <th>group_prior_var<\/th>\n      <th>enrichment<\/th>\n      <th>group_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>df_gene &lt;- process_gene_data(ctwas_res = ctwas_res)

df_gene &lt;- attributable_pip(df_gene,combined_pip_col = &quot;combined_pip&quot;)

DT::datatable(df_gene,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;top20 Genes sorted bu susiepip, 0 means pip=0; NA means no such gene in this tissue&#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-6ae57a2e1747c2662285" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-6ae57a2e1747c2662285">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">top20 Genes sorted bu susiepip, 0 means pip=0; NA means no such gene in this tissue<\/caption>","data":[["5638","6946","7843","9518","8207","1212","8737","9735","777","3871","972","7975","8445","2939","9064","6575","333","5263","544","1398"],["PAQR5","RP11-348J24.1","SLC35G1","ZDHHC21","STAM2","CCDC102A","TMEM74","ZNF57","BBS2","KB-208E9.1","C19orf54","SMOC2","TCOF1","FLOT1","U2AF1","RHOQ","ALDH1L1","NDUFS5","ARHGEF19","CDC42EP1"],[0.567122938002885,0.300944804677624,0.294699477084096,0.258123734719164,0.255559336160831,0.246217493503463,0.239496290297166,0.236279408814276,0.228317806188612,0.224555157829002,0.224400172147453,0.223538524046769,0.218724396406357,0.214809056425351,0.212140200145453,0.206523760438616,0.203483398407194,0.199945511091142,0.194675107382595,0.188002727344017],[0.373307992489503,0.300944804677624,0.217899966682143,0.177782694938747,0.136077996092127,0.138243923771926,0.1548382455649,0.236279408814276,0.106875625459874,0.196329422488174,0.126157692108307,0.206246129810668,0.218724396406357,0.0950175519886452,0.131756804675771,0.127491713235825,0.135130490220062,0.0780445686209869,0.0770893942065503,0.125844333195752],[65.82487984071,100,73.9397194858145,68.8749894046638,53.2471237937827,56.1470762311946,64.6516258655936,100,46.8100264469011,87.4303776347362,56.219962266967,92.2642442461134,100,44.2334944205041,62.1083625759911,61.7322253696415,66.4086069319768,39.0329186162182,39.5989991956427,66.9375040317769],[null,null,0.0289111030899879,0.0305918984493906,0.0174954580025822,0.0446275647253601,0.0359642104780558,null,0.0396822626855997,0.0158498259980061,0.0289893167820375,null,null,0.0327663440248865,null,0.0315172856639926,0.0399237099623311,0.0351449994889455,0.0364923146481684,0.0443170575728833],[null,null,9.81036796401839,11.851641028933,6.84594750691158,18.1252615686839,15.0166044047829,null,17.3802750420694,7.05832195138253,12.9185804558959,null,null,15.2537069759315,null,15.2608521155416,19.6201313103878,17.57728858085,18.7452392546778,23.5725609936443],[0.0890448092746062,null,null,null,0.101985882066122,0.0362160912405226,null,null,0.038120310642355,null,0.0431655147405575,null,null,0.0320910508405342,null,0.0142572380854444,null,0.040593195630375,0.0319501840675784,0.0178413365753819],[15.7011475480389,null,null,null,39.9069286993057,14.7089838033841,null,null,16.6961619326633,null,19.2359543789448,null,null,14.9393379285599,null,6.90343719055125,null,20.3021290194764,16.4120541640625,9.48993497457879],[0.0646486868837618,null,0.0478884073119655,0.0298644012925547,null,null,0.0262311781060077,null,0.0328742012062915,null,0.0116230620325466,null,null,0.0464941883990632,0.0478661346490588,0.027371877043456,0.0284291982248011,0.028588509307626,0.0295287216333582,null],[11.3994131698184,null,16.2499125501671,11.5698005551666,null,null,10.9526448503483,null,14.3984395063494,null,5.17961368804526,null,null,21.6444265306014,22.5634437113945,13.2536212711426,13.9712617576354,14.2981501068031,15.1682061617284,null],[0.0401214493550137,null,null,0.019884740038472,null,0.0271299137656545,0.0224626561482028,null,0.0107654061944923,0.0123759093428214,0.0144645864840042,0.0172923942361012,null,0.00843992117222192,0.0325172608206229,0.00588564640989819,null,0.0175742380432082,0.0196144928269396,null],[7.07455944143272,null,null,7.70356901123656,null,11.0186783967374,9.37912487927525,null,4.71509707201682,5.51130041388123,6.44588921014711,7.73575575388663,null,3.92903414440299,15.3281937126144,2.8498640531231,null,8.78951367665228,10.0755012238885,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genename<\/th>\n      <th>combined_pip<\/th>\n      <th>Heart_Left_Ventricle<\/th>\n      <th>Heart_Left_Ventricle_Attributable_PIP (%)<\/th>\n      <th>Artery_Coronary<\/th>\n      <th>Artery_Coronary_Attributable_PIP (%)<\/th>\n      <th>Kidney_Cortex<\/th>\n      <th>Kidney_Cortex_Attributable_PIP (%)<\/th>\n      <th>Liver<\/th>\n      <th>Liver_Attributable_PIP (%)<\/th>\n      <th>Lung<\/th>\n      <th>Lung_Attributable_PIP (%)<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7,8,9,10,11,12]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>attributable_columns &lt;- colnames(df_gene)[seq(4,12,by=2)]

plot_stacked_histogram(df_gene, &quot;genename&quot;, &quot;combined_pip&quot;, attributable_columns)</code></pre>
<p><img src="figure/multi_tissue.Rmd/unnamed-chunk-4-4.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>table_show &lt;- ctwas_res[ctwas_res$genename %in% df_gene$genename,]
table_show &lt;- rbind(table_show,ctwas_res[ctwas_res$type!=&quot;SNP&quot; &amp; ctwas_res$susie_pip &gt; 0.8,])
table_show &lt;- table_show[!duplicated(table_show$id),]
DT::datatable(table_show,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Detailed table for genes showed above&#39;),options = list(pageLength = 5))</code></pre>
<div id="htmlwidget-6f3f83fd1147bd8f7edb" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-6f3f83fd1147bd8f7edb">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Detailed table for genes showed above<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69"],[15,5,19,5,10,6,22,9,8,16,2,3,21,2,19,22,16,2,6,15,1,1,15,10,21,6,16,22,19,1,15,3,16,16,1,16,8,1,16,6,21,6,1,2,9,9,1,19,10,1,3,2,16,8,8,9,1,22,1,2,6,22,19,2,22,19,16,6,2],["ENSG00000137819.13|mashr_Heart_Left_Ventricle","ENSG00000248864.1|mashr_Heart_Left_Ventricle","ENSG00000171970.12|mashr_Heart_Left_Ventricle","ENSG00000070814.17|mashr_Heart_Left_Ventricle","ENSG00000176273.14|mashr_Heart_Left_Ventricle","ENSG00000112562.18|mashr_Heart_Left_Ventricle","ENSG00000272733.1|mashr_Heart_Left_Ventricle","ENSG00000175893.11|mashr_Heart_Left_Ventricle","ENSG00000164841.4|mashr_Heart_Left_Ventricle","ENSG00000135736.5|mashr_Heart_Left_Ventricle","ENSG00000115145.9|mashr_Heart_Left_Ventricle","ENSG00000144908.13|mashr_Heart_Left_Ventricle","ENSG00000160201.11|mashr_Heart_Left_Ventricle","ENSG00000119729.11|mashr_Heart_Left_Ventricle","ENSG00000188493.14|mashr_Heart_Left_Ventricle","ENSG00000128283.6|mashr_Heart_Left_Ventricle","ENSG00000125124.11|mashr_Heart_Left_Ventricle","ENSG00000115145.9|mashr_Kidney_Cortex","ENSG00000137312.14|mashr_Heart_Left_Ventricle","ENSG00000137819.13|mashr_Kidney_Cortex","ENSG00000168653.10|mashr_Heart_Left_Ventricle","ENSG00000142632.16|mashr_Heart_Left_Ventricle","ENSG00000137819.13|mashr_Liver","ENSG00000176273.14|mashr_Liver","ENSG00000160201.11|mashr_Liver","ENSG00000137312.14|mashr_Liver","ENSG00000135736.5|mashr_Artery_Coronary","ENSG00000128283.6|mashr_Artery_Coronary","ENSG00000188493.14|mashr_Kidney_Cortex","ENSG00000168653.10|mashr_Kidney_Cortex","ENSG00000137819.13|mashr_Lung","ENSG00000144908.13|mashr_Artery_Coronary","ENSG00000125124.11|mashr_Artery_Coronary","ENSG00000125124.11|mashr_Kidney_Cortex","ENSG00000142632.16|mashr_Artery_Coronary","ENSG00000135736.5|mashr_Kidney_Cortex","ENSG00000164841.4|mashr_Artery_Coronary","ENSG00000168653.10|mashr_Artery_Coronary","ENSG00000125124.11|mashr_Liver","ENSG00000137312.14|mashr_Artery_Coronary","ENSG00000160201.11|mashr_Lung","ENSG00000137312.14|mashr_Kidney_Cortex","ENSG00000142632.16|mashr_Kidney_Cortex","ENSG00000119729.11|mashr_Artery_Coronary","ENSG00000175893.11|mashr_Artery_Coronary","ENSG00000175893.11|mashr_Liver","ENSG00000142632.16|mashr_Liver","ENSG00000188493.14|mashr_Artery_Coronary","ENSG00000176273.14|mashr_Artery_Coronary","ENSG00000168653.10|mashr_Liver","ENSG00000144908.13|mashr_Liver","ENSG00000119729.11|mashr_Liver","ENSG00000135736.5|mashr_Lung","ENSG00000164841.4|mashr_Liver","ENSG00000164841.4|mashr_Lung","ENSG00000175893.11|mashr_Lung","ENSG00000142632.16|mashr_Lung","ENSG00000128283.6|mashr_Kidney_Cortex","ENSG00000168653.10|mashr_Lung","ENSG00000115145.9|mashr_Artery_Coronary","ENSG00000112562.18|mashr_Lung","ENSG00000272733.1|mashr_Artery_Coronary","ENSG00000188493.14|mashr_Lung","ENSG00000119729.11|mashr_Kidney_Cortex","ENSG00000272733.1|mashr_Lung","ENSG00000188493.14|mashr_Liver","ENSG00000125124.11|mashr_Lung","ENSG00000137312.14|mashr_Lung","ENSG00000119729.11|mashr_Lung"],[69299268,92329276,2897135,150356800,93893644,168441433,23686225,14672654,108763989,57536792,152174599,126198015,43028608,46541511,40751911,37560207,56520101,152176276,30742494,69299268,39026790,16210794,69299268,93893644,43031887,30745519,57536792,37560207,40751911,39026408,69299268,126197935,56520101,56520101,16210794,57536650,108787887,39026790,56520101,30742457,43031887,30742494,16210794,46541511,14665643,14692678,16210794,40746867,93891204,39026790,126191538,46541511,57536792,108788312,108787887,14665543,16200034,38117991,39026790,152176079,168446089,23631538,40751141,46541996,23635894,40751141,56513342,30742457,46541511],["Heart_Left_Ventricle","Heart_Left_Ventricle","Heart_Left_Ventricle","Heart_Left_Ventricle","Heart_Left_Ventricle","Heart_Left_Ventricle","Heart_Left_Ventricle","Heart_Left_Ventricle","Heart_Left_Ventricle","Heart_Left_Ventricle","Heart_Left_Ventricle","Heart_Left_Ventricle","Heart_Left_Ventricle","Heart_Left_Ventricle","Heart_Left_Ventricle","Heart_Left_Ventricle","Heart_Left_Ventricle","Kidney_Cortex","Heart_Left_Ventricle","Kidney_Cortex","Heart_Left_Ventricle","Heart_Left_Ventricle","Liver","Liver","Liver","Liver","Artery_Coronary","Artery_Coronary","Kidney_Cortex","Kidney_Cortex","Lung","Artery_Coronary","Artery_Coronary","Kidney_Cortex","Artery_Coronary","Kidney_Cortex","Artery_Coronary","Artery_Coronary","Liver","Artery_Coronary","Lung","Kidney_Cortex","Kidney_Cortex","Artery_Coronary","Artery_Coronary","Liver","Liver","Artery_Coronary","Artery_Coronary","Liver","Liver","Liver","Lung","Liver","Lung","Lung","Lung","Kidney_Cortex","Lung","Artery_Coronary","Lung","Artery_Coronary","Lung","Kidney_Cortex","Lung","Liver","Lung","Lung","Lung"],[15,5,19,5,10,6,22,9,8,16,2,3,21,2,19,22,16,2,6,15,1,1,15,10,21,6,16,22,19,1,15,3,16,16,1,16,8,1,16,6,21,6,1,2,9,9,1,19,10,1,3,2,16,8,8,9,1,22,1,2,6,22,19,2,22,19,16,6,2],[32,54,3,88,60,110,6,12,74,30,91,78,21,29,28,15,30,91,24,32,24,11,32,60,21,24,30,15,28,24,32,78,30,30,11,30,74,24,30,24,21,24,11,29,12,12,11,28,60,24,78,29,30,74,74,12,11,15,24,91,110,6,28,29,6,28,30,24,29],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0.373307992489503,0.300944804677624,0.236279408814276,0.218724396406357,0.217899966682143,0.206246129810668,0.196329422488174,0.177782694938747,0.1548382455649,0.138243923771926,0.136077996092127,0.135130490220062,0.131756804675771,0.127491713235825,0.126157692108307,0.125844333195752,0.106875625459874,0.101985882066122,0.0950175519886452,0.0890448092746062,0.0780445686209869,0.0770893942065503,0.0646486868837618,0.0478884073119655,0.0478661346490588,0.0464941883990632,0.0446275647253601,0.0443170575728833,0.0431655147405575,0.040593195630375,0.0401214493550137,0.0399237099623311,0.0396822626855997,0.038120310642355,0.0364923146481684,0.0362160912405226,0.0359642104780558,0.0351449994889455,0.0328742012062915,0.0327663440248865,0.0325172608206229,0.0320910508405342,0.0319501840675784,0.0315172856639926,0.0305918984493906,0.0298644012925547,0.0295287216333582,0.0289893167820375,0.0289111030899879,0.028588509307626,0.0284291982248011,0.027371877043456,0.0271299137656545,0.0262311781060077,0.0224626561482028,0.019884740038472,0.0196144928269396,0.0178413365753819,0.0175742380432082,0.0174954580025822,0.0172923942361012,0.0158498259980061,0.0144645864840042,0.0142572380854444,0.0123759093428214,0.0116230620325466,0.0107654061944923,0.00843992117222192,0.00588564640989819],[30.4463982502351,28.1481874506131,25.6876224942145,25.170742500085,25.5683220568868,24.8882556011454,24.4061157553413,23.7796859203656,22.8628658190983,21.6935852583532,22.2208560944052,21.8879461563063,21.2827892737944,25.4128247277824,20.4955930232786,20.4085675081848,19.6792524791324,14.5877895211229,20.576071420714,13.8010150587486,16.7523000747196,16.0445291843787,9.26097439036382,7.92551801406274,7.79017463743804,8.63821382152427,5.64008111661135,5.36069702049365,9.56546786053743,9.24763141126689,5.58776712888745,5.42224531227665,5.22646604880701,9.26563396339252,4.41188296531054,9.01392369975636,5.00760488846741,4.56576405405846,6.3238258826561,5.39711519951556,4.75798351114045,9.59972338488942,7.67087837336302,6.44870524439135,4.35280955356995,5.99067481990417,5.31821763291972,3.90149379332258,4.21934416574271,5.5119819594702,5.94835868385021,7.8829479738626,4.2358810860605,5.57905810629473,3.7865731515632,3.35441304453659,2.8424457266162,5.19281116303494,2.76070214396962,2.66521742442179,2.90691102013311,1.97452162886888,2.19817644169272,7.29819795816183,1.89340760457284,2.00225556171832,1.52041433684695,1.51858267365521,1.46526397257779],["PAQR5","RP11-348J24.1","ZNF57","TCOF1","SLC35G1","SMOC2","KB-208E9.1","ZDHHC21","TMEM74","CCDC102A","STAM2","ALDH1L1","U2AF1","RHOQ","C19orf54","CDC42EP1","BBS2","STAM2","FLOT1","PAQR5","NDUFS5","ARHGEF19","PAQR5","SLC35G1","U2AF1","FLOT1","CCDC102A","CDC42EP1","C19orf54","NDUFS5","PAQR5","ALDH1L1","BBS2","BBS2","ARHGEF19","CCDC102A","TMEM74","NDUFS5","BBS2","FLOT1","U2AF1","FLOT1","ARHGEF19","RHOQ","ZDHHC21","ZDHHC21","ARHGEF19","C19orf54","SLC35G1","NDUFS5","ALDH1L1","RHOQ","CCDC102A","TMEM74","TMEM74","ZDHHC21","ARHGEF19","CDC42EP1","NDUFS5","STAM2","SMOC2","KB-208E9.1","C19orf54","RHOQ","KB-208E9.1","C19orf54","BBS2","FLOT1","RHOQ"],[4.27739812347521,3.94396786139664,-3.40322511499351,3.42602204933122,3.53850562714332,-3.29310961970702,3.24740613754196,-3.03642681920226,3.29571095044501,-2.91646835547162,2.97246561055579,3.00493025999773,-2.93896893987011,-3.19669491967949,2.81387964444068,2.71964554618432,-2.90670363413705,3.83662214970145,3.07195225515944,-4.27739812347521,-2.42511077238344,-2.33315857626847,-4.27739812347521,-3.53850562714332,-3.18732131173182,-3.26162891479058,-2.91646835547162,2.71964554618432,2.81387964444068,-2.68181426119663,-4.27739812347521,2.76493555098672,-2.91553538704071,-2.91553538704071,-2.33315857626847,-2.72439732400598,-2.58884163623782,-2.42511077238344,-2.90975117016531,3.07195225516082,-3.18732131173182,3.07195225515944,-2.33315857626847,-3.19669491967949,-2.36030133560721,-2.60933623201017,-2.33315857626847,2.21081710284771,2.73864110757452,-2.42511077238344,2.6959229711091,-3.19669491967949,-2.91646835547162,-2.43191764728222,-2.58884163623782,-2.37837435403357,-2.01913318850421,-1.72617738836482,-1.98129463287158,-1.79045036058842,-2.20956903343255,-0.750863151640919,1.47237610415857,1.98421486116893,1.21285106647241,0.279238626398419,-0.746421603665683,0.633741361528387,-0.361103697203873]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chrom<\/th>\n      <th>id<\/th>\n      <th>pos<\/th>\n      <th>type<\/th>\n      <th>region_tag1<\/th>\n      <th>region_tag2<\/th>\n      <th>cs_index<\/th>\n      <th>susie_pip<\/th>\n      <th>mu2<\/th>\n      <th>genename<\/th>\n      <th>z<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,3,5,6,7,8,9,11]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="pulse" class="section level1">
<h1>Pulse</h1>
<pre class="r"><code>outputdir &lt;- &quot;/project/xinhe/xsun/ctwas/2.TIBETN/output/multi/Pulse/&quot;
outname &lt;- &quot;Pulse_ctwas&quot;
gwas_n &lt;- 966
weight &lt;- &quot;Heart_Atrial_Appendage,Kidney_Cortex,Artery_Tibial,Whole_Blood,Liver&quot;

load(paste0(outputdir,outname,&quot;_z_snp.Rd&quot;))
load(paste0(outputdir,outname,&quot;_z_gene.Rd&quot;))

ctwas_parameters &lt;- ctwas:::ctwas_summarize_parameters(outputdir = outputdir,
                                               outname = outname,
                                               gwas_n = gwas_n,
                                               thin = thin)

ctwas_parameters$convergence_plot</code></pre>
<p><img src="figure/multi_tissue.Rmd/unnamed-chunk-5-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ctwas_res &lt;- process_data(weight = weight,outputdir = outputdir,outname = outname,z_snp = z_snp,z_gene = z_gene)
para &lt;- process_ctwas_results(ctwas_parameters = ctwas_parameters,ctwas_res = ctwas_res)
DT::datatable(para,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Parameters&#39;))</code></pre>
<div id="htmlwidget-57ddc594ddb93f271c91" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-57ddc594ddb93f271c91">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Parameters<\/caption>","data":[["1","2","3","4","5","6"],["Artery_Tibial","Heart_Atrial_Appendage","Kidney_Cortex","Liver","SNP","Whole_Blood"],[0.0998850428157301,0.25345657088703,0.370261600152272,0.0510315132493902,null,0.141080738453597],["5677","5180","3353","4259","2694600","4774"],["0.00430854938934982","0.00404043532247074","0.00514834164702654","0.00310586985947092","0.000307693393931392","0.00375438380188699"],["0.601806293567781","1.53255473133729","2.20982021133908","0.382802439158904","0.615519818884877","0.515241660606616"],["14.0027360818495","13.1313684406616","16.73205128406","10.094041408518",null,"12.2017042807364"],["0.0152380561192168","0.0332044878252535","0.0394894175020503","0.00524189677242496","0.528296085113238","0.00955993472640351"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>type<\/th>\n      <th>max_susie_pip<\/th>\n      <th>group_size<\/th>\n      <th>group_prior<\/th>\n      <th>group_prior_var<\/th>\n      <th>enrichment<\/th>\n      <th>group_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>df_gene &lt;- process_gene_data(ctwas_res = ctwas_res)

df_gene &lt;- attributable_pip(df_gene,combined_pip_col = &quot;combined_pip&quot;)

DT::datatable(df_gene,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;top20 Genes sorted bu susiepip, 0 means pip=0; NA means no such gene in this tissue&#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-0a12cbaf0b5628a2753a" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0a12cbaf0b5628a2753a">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">top20 Genes sorted bu susiepip, 0 means pip=0; NA means no such gene in this tissue<\/caption>","data":[["4774","2624","3754","8946","8043","1057","9425","9532","4832","4903","4895","1141","9530","866","1329","3396","1087","7095","2115","413"],["LOXL2","EFNA5","HIF3A","TBX2","SELENOT","C17orf62","TRIM22","TTC21B","LRRC8B","LYRM2","LYNX1","C2orf74","TTC17","BCKDHB","CBX5","GK5","C1RL","RNF150","CTD-2054N24.2","AMOTL1"],[0.523610135157543,0.402830775201078,0.335851738786359,0.321195787248804,0.305019801991914,0.285609407628198,0.281725999381933,0.276735157208605,0.269875332212086,0.256813270860002,0.255932185918719,0.254194876244534,0.247600702585518,0.247364390074011,0.245837517853831,0.23790722012534,0.235751281834069,0.23548179258037,0.226063652335024,0.218928028310784],[0.2853537376762,0.370261600152272,0.0982169679843135,0.130432187985166,null,0.0725738152272328,0.109874839587633,0.175250413877612,null,0.0657405929739791,0.0825525920695051,0.126377503586566,0.147931008981364,0.0138451800616627,null,null,0.136112001776539,0.074950648224077,0.0939452339517851,0.0841012806484857],[54.4973671279956,91.9149238206668,29.2441445559378,40.6083121769375,null,25.4101627218485,39.0006033623744,63.3278458889512,null,25.5985965031444,32.2556507588783,49.7167785022511,59.7457953215099,5.5970788913959,null,null,57.7354238405966,31.8286383854906,41.556983168864,38.4150358898303],[0.179723732176078,0.0148634179448101,0.12692743654346,0.0886889846485557,0.25345657088703,0.102731721768441,0.0745391027116521,null,0.148504205958974,0.0813533387517715,0.0246199844057218,null,0.0996696936041539,0.113037791802569,0.118617086115863,0.220318968251647,0.0194891506803737,0.101406437568003,0.0659943806010994,0.0603170945472562],[34.3239597762948,3.68974240793565,37.7926989457097,27.6121257405707,83.0951201305118,35.9693059908501,26.4580134155812,null,55.0269655035642,31.678011996553,9.61972966289627,null,40.2542046784901,45.6968732519455,48.2501967768768,92.607096218254,8.26682702581907,43.0633878130484,29.1928312753686,27.5511066411431],[0.0585326653052648,null,0.0384123424259643,0.0149999780097018,null,0.0485095806830188,0.0373733596954921,null,0.0267719394131817,0.0143361446321262,0.035631436985287,0.0414953961603514,null,0.0211893397709535,0.0450795210403658,0.017588251873693,0.0296635268775977,null,0.0164638831139189,0.0161264015287204],[11.1786730957096,null,11.4372915158254,4.67004195110522,null,16.9845878277818,13.2658539777954,null,9.92011355529985,5.58232235589622,13.9222180506063,16.3242457021176,null,8.56604289914715,18.3371201572125,7.39290378174598,12.5825516819357,null,7.28285283541275,7.36607443695048],[null,0.0177057571039955,0.0474334800261377,0.0557446386244979,0.0263611061905943,0.0617942899495058,0.0468372886098191,0.0659622619934632,0.0796705683990786,0.0718241495539249,0.0433589236877369,0.0551281356215305,null,0.0645254683089278,0.0637608316048674,null,0.0306062570387581,0.05912470678829,0.0372540856514388,0.0445794536398543],[null,4.39533377139755,14.123339125039,17.3553455049886,8.64242453061888,21.6359434595196,16.6251211150457,23.8358807239444,29.5212488470299,27.9674602926103,16.9415673656252,21.6873512306746,null,26.0851888542331,25.9361679866855,null,12.9824350479248,25.107973801461,16.4794672945603,20.3626068273773],[null,null,0.0248615118064838,0.0313299979808822,0.0252021249142894,null,0.0131014087773366,0.0355224813375297,0.0149286184408522,0.0235590449482003,0.0262163304196806,0.0311938408760856,null,0.0347666101298983,0.0183800790927346,null,0.0198803454608004,null,0.0124060690167817,0.0138037979464678],[null,null,7.40252585748815,9.75417462639803,8.26245533886929,null,4.65040812920328,12.8362733871044,5.53167209410613,9.17360885179612,10.2434675519892,12.2716245649568,null,14.0548161032783,7.4765150792252,null,8.43276240372384,null,5.48786542579434,6.30517620469879]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genename<\/th>\n      <th>combined_pip<\/th>\n      <th>Kidney_Cortex<\/th>\n      <th>Kidney_Cortex_Attributable_PIP (%)<\/th>\n      <th>Heart_Atrial_Appendage<\/th>\n      <th>Heart_Atrial_Appendage_Attributable_PIP (%)<\/th>\n      <th>Whole_Blood<\/th>\n      <th>Whole_Blood_Attributable_PIP (%)<\/th>\n      <th>Artery_Tibial<\/th>\n      <th>Artery_Tibial_Attributable_PIP (%)<\/th>\n      <th>Liver<\/th>\n      <th>Liver_Attributable_PIP (%)<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7,8,9,10,11,12]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>attributable_columns &lt;- colnames(df_gene)[seq(4,12,by=2)]

plot_stacked_histogram(df_gene, &quot;genename&quot;, &quot;combined_pip&quot;, attributable_columns)</code></pre>
<p><img src="figure/multi_tissue.Rmd/unnamed-chunk-5-4.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>table_show &lt;- ctwas_res[ctwas_res$genename %in% df_gene$genename,]
table_show &lt;- rbind(table_show,ctwas_res[ctwas_res$type!=&quot;SNP&quot; &amp; ctwas_res$susie_pip &gt; 0.8,])
table_show &lt;- table_show[!duplicated(table_show$id),]
DT::datatable(table_show,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Detailed table for genes showed above&#39;),options = list(pageLength = 5))</code></pre>
<div id="htmlwidget-96697933f1d703263334" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-96697933f1d703263334">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Detailed table for genes showed above<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81"],[5,8,3,3,8,2,1,11,12,17,19,2,12,6,11,17,4,11,19,15,17,11,8,6,1,4,11,17,6,15,2,6,6,12,17,11,4,8,17,2,17,19,11,12,11,8,8,2,19,11,15,8,2,6,17,2,12,12,1,3,8,3,19,8,6,6,12,12,12,5,3,15,11,17,1,5,6,6,11,11,15],["ENSG00000184349.12|mashr_Kidney_Cortex","ENSG00000134013.15|mashr_Kidney_Cortex","ENSG00000198843.12|mashr_Heart_Atrial_Appendage","ENSG00000175066.15|mashr_Heart_Atrial_Appendage","ENSG00000134013.15|mashr_Heart_Atrial_Appendage","ENSG00000123607.14|mashr_Kidney_Cortex","ENSG00000197147.13|mashr_Heart_Atrial_Appendage","ENSG00000052841.14|mashr_Kidney_Cortex","ENSG00000139178.10|mashr_Kidney_Cortex","ENSG00000121068.13|mashr_Kidney_Cortex","ENSG00000124440.15|mashr_Heart_Atrial_Appendage","ENSG00000237651.6|mashr_Kidney_Cortex","ENSG00000094916.15|mashr_Heart_Atrial_Appendage","ENSG00000083123.14|mashr_Heart_Atrial_Appendage","ENSG00000132274.15|mashr_Kidney_Cortex","ENSG00000178927.17|mashr_Heart_Atrial_Appendage","ENSG00000170153.10|mashr_Heart_Atrial_Appendage","ENSG00000052841.14|mashr_Heart_Atrial_Appendage","ENSG00000124440.15|mashr_Kidney_Cortex","ENSG00000259363.5|mashr_Kidney_Cortex","ENSG00000121068.13|mashr_Heart_Atrial_Appendage","ENSG00000166025.17|mashr_Kidney_Cortex","ENSG00000180155.19|mashr_Kidney_Cortex","ENSG00000083099.10|mashr_Heart_Atrial_Appendage","ENSG00000197147.13|mashr_Artery_Tibial","ENSG00000170153.10|mashr_Kidney_Cortex","ENSG00000132274.15|mashr_Heart_Atrial_Appendage","ENSG00000178927.17|mashr_Kidney_Cortex","ENSG00000083099.10|mashr_Artery_Tibial","ENSG00000259363.5|mashr_Heart_Atrial_Appendage","ENSG00000123607.14|mashr_Artery_Tibial","ENSG00000083099.10|mashr_Kidney_Cortex","ENSG00000083123.14|mashr_Artery_Tibial","ENSG00000094916.15|mashr_Artery_Tibial","ENSG00000178927.17|mashr_Artery_Tibial","ENSG00000166025.17|mashr_Heart_Atrial_Appendage","ENSG00000170153.10|mashr_Artery_Tibial","ENSG00000134013.15|mashr_Whole_Blood","ENSG00000121068.13|mashr_Artery_Tibial","ENSG00000237651.6|mashr_Artery_Tibial","ENSG00000178927.17|mashr_Whole_Blood","ENSG00000124440.15|mashr_Artery_Tibial","ENSG00000132274.15|mashr_Artery_Tibial","ENSG00000094916.15|mashr_Whole_Blood","ENSG00000166025.17|mashr_Artery_Tibial","ENSG00000283992.1|mashr_Artery_Tibial","ENSG00000180155.19|mashr_Artery_Tibial","ENSG00000237651.6|mashr_Whole_Blood","ENSG00000124440.15|mashr_Whole_Blood","ENSG00000132274.15|mashr_Whole_Blood","ENSG00000259363.5|mashr_Artery_Tibial","ENSG00000180155.19|mashr_Whole_Blood","ENSG00000123607.14|mashr_Liver","ENSG00000083123.14|mashr_Liver","ENSG00000121068.13|mashr_Liver","ENSG00000237651.6|mashr_Liver","ENSG00000139178.10|mashr_Artery_Tibial","ENSG00000139178.10|mashr_Whole_Blood","ENSG00000197147.13|mashr_Whole_Blood","ENSG00000198843.12|mashr_Artery_Tibial","ENSG00000180155.19|mashr_Liver","ENSG00000198843.12|mashr_Liver","ENSG00000124440.15|mashr_Liver","ENSG00000283992.1|mashr_Heart_Atrial_Appendage","ENSG00000083099.10|mashr_Liver","ENSG00000083123.14|mashr_Whole_Blood","ENSG00000139178.10|mashr_Liver","ENSG00000139178.10|mashr_Heart_Atrial_Appendage","ENSG00000094916.15|mashr_Liver","ENSG00000184349.12|mashr_Artery_Tibial","ENSG00000175066.15|mashr_Whole_Blood","ENSG00000259363.5|mashr_Whole_Blood","ENSG00000166025.17|mashr_Whole_Blood","ENSG00000121068.13|mashr_Whole_Blood","ENSG00000197147.13|mashr_Liver","ENSG00000184349.12|mashr_Heart_Atrial_Appendage","ENSG00000083099.10|mashr_Whole_Blood","ENSG00000083123.14|mashr_Kidney_Cortex","ENSG00000166025.17|mashr_Liver","ENSG00000132274.15|mashr_Liver","ENSG00000259363.5|mashr_Liver"],[107648949,23425787,150601897,142226081,23425787,165954060,89522381,43357956,7109428,61399054,46296578,61145372,54281329,80106428,5689666,82450472,141212652,43357956,46296611,99871875,61399054,94706514,142778078,89630557,89547585,141213673,5689666,82427889,89630557,99871875,165954060,89634437,80106428,54281329,82450543,94706514,141212743,23421661,61399054,61145372,82450102,46296611,5689666,54280451,94706514,142778078,142780034,61144588,46296611,5689666,99824037,142778078,165954060,80106428,61399054,61145372,7100169,7097002,89557240,150602660,142778078,150602502,46296611,142769957,89636884,80107031,7100169,7107010,54280451,107425435,142214459,99791077,94706829,61399531,89525438,107671300,89629144,80106428,94706829,5689947,99807654],["Kidney_Cortex","Kidney_Cortex","Heart_Atrial_Appendage","Heart_Atrial_Appendage","Heart_Atrial_Appendage","Kidney_Cortex","Heart_Atrial_Appendage","Kidney_Cortex","Kidney_Cortex","Kidney_Cortex","Heart_Atrial_Appendage","Kidney_Cortex","Heart_Atrial_Appendage","Heart_Atrial_Appendage","Kidney_Cortex","Heart_Atrial_Appendage","Heart_Atrial_Appendage","Heart_Atrial_Appendage","Kidney_Cortex","Kidney_Cortex","Heart_Atrial_Appendage","Kidney_Cortex","Kidney_Cortex","Heart_Atrial_Appendage","Artery_Tibial","Kidney_Cortex","Heart_Atrial_Appendage","Kidney_Cortex","Artery_Tibial","Heart_Atrial_Appendage","Artery_Tibial","Kidney_Cortex","Artery_Tibial","Artery_Tibial","Artery_Tibial","Heart_Atrial_Appendage","Artery_Tibial","Whole_Blood","Artery_Tibial","Artery_Tibial","Whole_Blood","Artery_Tibial","Artery_Tibial","Whole_Blood","Artery_Tibial","Artery_Tibial","Artery_Tibial","Whole_Blood","Whole_Blood","Whole_Blood","Artery_Tibial","Whole_Blood","Liver","Liver","Liver","Liver","Artery_Tibial","Whole_Blood","Whole_Blood","Artery_Tibial","Liver","Liver","Liver","Heart_Atrial_Appendage","Liver","Whole_Blood","Liver","Heart_Atrial_Appendage","Liver","Artery_Tibial","Whole_Blood","Whole_Blood","Whole_Blood","Whole_Blood","Liver","Heart_Atrial_Appendage","Whole_Blood","Kidney_Cortex","Liver","Liver","Liver"],[5,8,3,3,8,2,1,11,12,17,19,2,12,6,11,17,4,11,19,15,17,11,8,6,1,4,11,17,6,15,2,6,6,12,17,11,4,8,17,2,17,19,11,12,11,8,8,2,19,11,15,8,2,6,17,2,12,12,1,3,8,3,19,8,6,6,12,12,12,5,3,15,11,17,1,5,6,6,11,11,15],[63,24,93,87,24,100,54,27,7,36,32,40,33,55,4,47,93,27,32,48,36,54,93,61,54,93,4,47,61,48,100,61,55,33,47,54,93,24,36,40,47,32,4,33,54,93,93,40,32,4,48,93,100,55,36,40,7,7,54,93,93,93,32,93,61,55,7,7,33,63,87,48,54,36,54,63,61,55,54,4,48],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0.370261600152272,0.2853537376762,0.25345657088703,0.220318968251647,0.179723732176078,0.175250413877612,0.148504205958974,0.147931008981364,0.136112001776539,0.130432187985166,0.12692743654346,0.126377503586566,0.118617086115863,0.113037791802569,0.109874839587633,0.102731721768441,0.101406437568003,0.0996696936041539,0.0982169679843135,0.0939452339517851,0.0886889846485557,0.0841012806484857,0.0825525920695051,0.0813533387517715,0.0796705683990786,0.074950648224077,0.0745391027116521,0.0725738152272328,0.0718241495539249,0.0659943806010994,0.0659622619934632,0.0657405929739791,0.0645254683089278,0.0637608316048674,0.0617942899495058,0.0603170945472562,0.05912470678829,0.0585326653052648,0.0557446386244979,0.0551281356215305,0.0485095806830188,0.0474334800261377,0.0468372886098191,0.0450795210403658,0.0445794536398543,0.0435529183507873,0.0433589236877369,0.0414953961603514,0.0384123424259643,0.0373733596954921,0.0372540856514388,0.035631436985287,0.0355224813375297,0.0347666101298983,0.0313299979808822,0.0311938408760856,0.0306062570387581,0.0296635268775977,0.0267719394131817,0.0263611061905943,0.0262163304196806,0.0252021249142894,0.0248615118064838,0.0246199844057218,0.0235590449482003,0.0211893397709535,0.0198803454608004,0.0194891506803737,0.0183800790927346,0.0177057571039955,0.017588251873693,0.0164638831139189,0.0161264015287204,0.0149999780097018,0.0149286184408522,0.0148634179448101,0.0143361446321262,0.0138451800616627,0.0138037979464678,0.0131014087773366,0.0124060690167817],[27.5782353491858,25.6625338222532,23.0311440806937,21.5700845283602,20.1933588821568,21.5534482205171,18.6067512387851,20.0814656242824,19.9568580981465,19.3822001058617,17.6434296409974,19.0476186969091,17.9927243879199,16.9880667238277,19.3402831724959,16.2449384256627,16.6057367427938,15.8813547502629,17.3676114424561,17.1676474710267,15.3410933347479,15.8827860268998,15.9504887571223,14.8118013116095,7.99681181670355,15.8798671025837,15.3087077799656,15.147170244415,7.6107226487907,13.6301018591059,7.27223213216356,14.4858475171351,7.26045229995723,7.64770749600292,7.00824552079272,12.6374015982489,7.11307237288104,6.7269163307767,6.62561017863982,6.52596809590899,5.87469062799598,6.02564316046015,6.61312684743702,6.08988793492911,5.58344555790696,5.60360818758521,5.58653402736047,5.29493318133308,5.09731875859453,5.57818856703882,5.16700624764437,4.75160782108713,4.31250531835627,4.30610981451349,3.96144147329056,3.90734381622338,4.47755917120791,4.31850578683351,3.8279832263791,4.30796752884519,3.40657638297118,3.75105185715756,3.32840173576869,7.30155620650702,3.16896223694039,3.09742501208695,2.79522210514683,6.22804823128386,2.84226855423471,2.35674773957797,2.59596112043917,2.24456745531189,1.92987052802014,1.84955429995682,1.88455543083021,4.50489125109773,1.70141308345275,3.73094905803761,1.53637850768386,2.01036978802681,1.43776068524732],["EFNA5","LOXL2","SELENOT","GK5","LOXL2","TTC21B","LRRC8B","TTC17","C1RL","TBX2","HIF3A","C2orf74","CBX5","BCKDHB","TRIM22","C17orf62","RNF150","TTC17","HIF3A","CTD-2054N24.2","TBX2","AMOTL1","LYNX1","LYRM2","LRRC8B","RNF150","TRIM22","C17orf62","LYRM2","CTD-2054N24.2","TTC21B","LYRM2","BCKDHB","CBX5","C17orf62","AMOTL1","RNF150","LOXL2","TBX2","C2orf74","C17orf62","HIF3A","TRIM22","CBX5","AMOTL1","LYNX1","LYNX1","C2orf74","HIF3A","TRIM22","CTD-2054N24.2","LYNX1","TTC21B","BCKDHB","TBX2","C2orf74","C1RL","C1RL","LRRC8B","SELENOT","LYNX1","SELENOT","HIF3A","LYNX1","LYRM2","BCKDHB","C1RL","C1RL","CBX5","EFNA5","GK5","CTD-2054N24.2","AMOTL1","TBX2","LRRC8B","EFNA5","LYRM2","BCKDHB","AMOTL1","TRIM22","CTD-2054N24.2"],[4.06198865902755,4.1430084861171,4.1703417880437,3.95003079870949,-4.1430084861171,-3.34939799496827,3.41303110248164,-3.09790884722423,-2.9482955075241,3.02952617292283,-3.2576950000955,-3.11988721255725,-3.10623259376961,-3.3989348911894,-2.97504839353322,3.13612498424058,-3.1927421221554,3.09790884722423,-2.85844540849163,2.68804264410221,3.02952617292283,2.50956273703831,2.60206890556664,-2.85332598883815,3.43772338704519,2.75512862835223,-2.97504839353322,2.60945052991403,-3.18711086855675,2.68804264410221,-3.34939799496827,-2.39795634113504,-3.3989348911894,-3.10623259376961,3.14514290471652,2.50956273703831,-3.19274212215509,-3.55428736224247,3.02952617292283,-3.11988721255725,3.1361249842403,-2.85844540849163,-2.97504839353322,2.9852052815744,2.50956273703831,2.60206890556664,2.59665158441046,-3.03914714106814,-2.85844540849163,-2.9449768042207,2.40812018931892,2.60206890556664,-3.34939799496827,-3.3989348911894,3.02952617292283,-3.11988721255725,-2.14667839888376,-2.31428268928113,2.36866558143453,2.24586254148187,2.60206890556664,3.04318081249811,-2.66832025225347,1.76913404069374,-2.39754528704518,-1.97060623542621,-2.14667839888376,-1.55716934130323,2.10543463995696,0.885878515099784,1.11473376984288,1.04669491894404,0.655078679439859,-0.473856866715845,-1.38789812702214,-0.772199205628722,-0.180104326351431,0.124023715061577,0.655078679439859,1.43345251504168,0.43677161308495]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chrom<\/th>\n      <th>id<\/th>\n      <th>pos<\/th>\n      <th>type<\/th>\n      <th>region_tag1<\/th>\n      <th>region_tag2<\/th>\n      <th>cs_index<\/th>\n      <th>susie_pip<\/th>\n      <th>mu2<\/th>\n      <th>genename<\/th>\n      <th>z<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,3,5,6,7,8,9,11]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
[1] C

attached base packages:
[1] tools     stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] cowplot_1.1.1     ggplot2_3.3.5     tidyr_1.3.0       dplyr_1.1.4      
[5] DT_0.22           RSQLite_2.3.1     data.table_1.14.2 ctwas_0.1.40     

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3      lattice_0.20-45   rprojroot_2.0.3   digest_0.6.29    
 [5] foreach_1.5.2     utf8_1.2.2        R6_2.5.1          evaluate_0.15    
 [9] highr_0.9         pillar_1.9.0      rlang_1.1.2       rstudioapi_0.13  
[13] whisker_0.4       jquerylib_0.1.4   blob_1.2.3        Matrix_1.5-3     
[17] rmarkdown_2.14    labeling_0.4.2    stringr_1.5.1     htmlwidgets_1.5.4
[21] bit_4.0.4         munsell_0.5.0     compiler_4.2.0    httpuv_1.6.5     
[25] xfun_0.30         pkgconfig_2.0.3   htmltools_0.5.2   tidyselect_1.2.0 
[29] tibble_3.2.1      workflowr_1.7.0   logging_0.10-108  codetools_0.2-18 
[33] fansi_1.0.3       withr_2.5.0       later_1.3.0       grid_4.2.0       
[37] jsonlite_1.8.0    gtable_0.3.0      lifecycle_1.0.4   DBI_1.1.2        
[41] git2r_0.30.1      magrittr_2.0.3    scales_1.2.0      cli_3.6.1        
[45] stringi_1.7.6     cachem_1.0.6      farver_2.1.0      fs_1.5.2         
[49] promises_1.2.0.1  pgenlibr_0.3.3    bslib_0.3.1       generics_0.1.2   
[53] vctrs_0.6.5       iterators_1.0.14  bit64_4.0.5       glue_1.6.2       
[57] purrr_1.0.2       crosstalk_1.2.0   fastmap_1.1.0     yaml_2.3.5       
[61] colorspace_2.0-3  memoise_2.0.1     knitr_1.39        sass_0.4.1       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
