---
title: "Multi tissue analysis -- using uncorrelated tissues reported by single tissue analysis (merging the LD blocks)"
author: "XSun"
date: "2024-01-01"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---

# Overview

## Traits

LB, MC, OxHb, Pulse

## Tissues 

We used the top 5 tissues decided [here](https://xsun1229.github.io/ctwas_tibetan/single_tissue_ss200.html)

The weights contains only protein-coding genes

## Settings

1. preharmonize snp z score:
harmonize_z=T, 
strand_ambig_action_z="drop"

2. impute gene z-scores for both sets of prediction weights by chromosome
harmonize_z = F (we have pre-harmonized)
harmonize_wgt = T
strand_ambig_action_wgt="drop"

3. ctwas_rss parameter estimation & fine mapping
LD merge=T, 
group_prior_var_structure = "shared"


## Functions used

```{r }
suppressMessages(library(ctwas))
suppressMessages(library(data.table))
suppressMessages(library(RSQLite))
suppressMessages(library(DT))
suppressMessages(library(tools))
suppressMessages(library(dplyr))
suppressMessages(library(tidyr))

thin <- 0.1

load("/project2/xinhe/shared_data/multigroup_ctwas/weights/gene_mapping_pred_nolnc.rdata")

process_data <- function(weight, outputdir, outname, z_snp, z_gene, display_datatable = TRUE) {
  # Splitting and constructing weight file paths
  weight <- unlist(strsplit(weight, split=","))
  weight <- paste0("/project2/xinhe/shared_data/multigroup_ctwas/weights/predictdb_nolnc/mashr_", weight, "_nolnc.db")

  # Reading cTWAS results
  ctwas_res <- fread(paste0(outputdir, outname, ".susieIrss.txt"))

  # Initialize gene_info dataframe
  gene_info <- data.frame(gene=character(), genename=character(), gene_type=character(), weight=character())
  for (i in seq_along(weight)) {
    sqlite <- dbDriver("SQLite")
    db <- dbConnect(sqlite, weight[i])
    query <- function(...) dbGetQuery(db, ...)
    gene_info_current <- query("SELECT gene, genename, gene_type FROM extra")
    dbDisconnect(db)

    gene_info_current$weight <- weight[i]
    gene_info <- rbind(gene_info, gene_info_current)
  }

  # Processing gene_info
  gene_info$weight <- sapply(gene_info$weight, function(x) rev(unlist(strsplit(file_path_sans_ext(x), "/")))[1])
  gene_info$id <- paste(gene_info$gene, gene_info$weight, sep="|")

  # Adding gene names to cTWAS results
  ctwas_res$genename[ctwas_res$type != "SNP"] <- gene_info$genename[match(ctwas_res$id[ctwas_res$type != "SNP"], gene_info$id)]

  # Adding z values
  ctwas_res$z[ctwas_res$type == "SNP"] <- z_snp$z[match(ctwas_res$id[ctwas_res$type == "SNP"], z_snp$id)]
  ctwas_res$z[ctwas_res$type != "SNP"] <- z_gene$z[match(ctwas_res$id[ctwas_res$type != "SNP"], z_gene$id)]

  # Sorting and displaying datatable if required
  ctwas_res <- ctwas_res[order(-ctwas_res$susie_pip),]
  if (display_datatable) {
    datatable(ctwas_res[ctwas_res$type != "SNP" & ctwas_res$susie_pip > 0.8,])
  }
  
  return(ctwas_res)
}

process_gene_data <- function(ctwas_res, weight = NULL) {
  # Aggregate susie_pip
  df_gene <- aggregate(ctwas_res$susie_pip[ctwas_res$type != "SNP"], 
                       by = list(ctwas_res$genename[ctwas_res$type != "SNP"]), 
                       FUN = sum)
  colnames(df_gene) <- c("genename", "combined_pip")

  # Optional part for dropping duplicated gene names
  # Uncomment this part if needed
  # df_gene <- df_gene[!(df_gene$genename %in% names(which(table(ctwas_res$genename) > length(weight)))),]

  # Collect tissue-level results
  all_tissue_names <- unique(ctwas_res$type[ctwas_res$type != "SNP"])

  df_gene_pips <- matrix(NA, nrow = nrow(df_gene), ncol = length(all_tissue_names))
  colnames(df_gene_pips) <- all_tissue_names

  ctwas_gene_res <- ctwas_res[ctwas_res$type != "SNP",]

  for (i in 1:nrow(df_gene)) {
    gene <- df_gene$genename[i]
    ctwas_gene_res_subset <- ctwas_gene_res[ctwas_gene_res$genename == gene,]
    df_gene_pips[i, ctwas_gene_res_subset$type] <- ctwas_gene_res_subset$susie_pip
  }

  df_gene <- cbind(df_gene, df_gene_pips)

  # Sort by combined PIP
  df_gene <- df_gene[order(-df_gene$combined_pip),]
  df_gene <- df_gene[, apply(df_gene, 2, function(x) {!all(is.na(x))})]  # Drop genes not imputed in any tissue

  # Determine number of rows to display
  n_rows <- max(sum(df_gene$combined_pip > 0.8), 20)

  # Select top rows based on the criteria
  df_to_display <- head(df_gene, n_rows)
  
  # Return a datatable
 # return(DT::datatable(df_to_display,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Genes with combined pip > 0.8, 0 means pip=0; NA means no pip reported by ctwas or no such gene in this tissue'),options = list(pageLength = 10) ))
    datatable_display <- DT::datatable(df_to_display, 
                                     caption = htmltools::tags$caption(style = 'caption-side: left; text-align: left; color:black; font-size:150%;', 
                                     'Genes with combined pip > 0.8, 0 means pip=0; NA means no pip reported by ctwas or no such gene in this tissue'), 
                                     options = list(pageLength = 10))
  return(df_to_display)
}

process_ctwas_results <- function(ctwas_parameters, ctwas_res) {
    # Process parameters
    para <- cbind(ctwas_parameters$group_size,
                  ctwas_parameters$group_prior,
                  ctwas_parameters$group_prior_var,
                  c(NA, ctwas_parameters$enrichment),
                  ctwas_parameters$group_pve)
    para <- as.data.frame(cbind(rownames(para), para))
    colnames(para) <- c("group","group_size","group_prior","group_prior_var","enrichment","group_pve")

    # Filter ctwas results for genes and high PIP
    ctwas_res_gene <- ctwas_res[ctwas_res$type != "SNP",]
    # ctwas_res_gene_pip08 <- ctwas_res_gene[ctwas_res_gene$susie_pip > 0.8,]
    # 
    # # Create summary table
    # summary <- as.data.frame(table(ctwas_res_gene_pip08$type))
    # colnames(summary) <- c("weights", "#of genes with susiepip >0.8")

    max_susie_pip_by_type <- ctwas_res_gene %>%
    group_by(type) %>%
    summarize(max_susie_pip = max(susie_pip, na.rm = TRUE))

    
    # Merge summary with parameters
    summary <- merge(max_susie_pip_by_type, para, by.x = "type", by.y = "group", all.y = TRUE)

    return(summary)
}

attributable_pip <- function(df, combined_pip_col) {
  # Identify the relevant columns (those ending with '_nolnc')
  relevant_cols <- names(df)[grepl("_nolnc$", names(df))]

  # Create and rearrange columns dynamically
  df <- df %>%
    mutate(across(all_of(relevant_cols), 
                  ~ .x / df[[combined_pip_col]] *100, 
                  .names = "{.col}_Attributable_PIP (%)")) %>%
    {
      # Building the new column order
      new_order <- unlist(lapply(relevant_cols, function(col) {
        c(col, paste0(col, "_Attributable_PIP (%)"))
      }))
      dplyr::select(., all_of(c("genename", combined_pip_col, new_order)))
    }

  df
}


plot_stacked_histogram <- function(df, genename_col, combined_pip_col, attributable_cols) {
  
  df[[genename_col]] <- factor(df[[genename_col]], levels = df[[genename_col]])
  
  long_df_attributable <- df %>%
    dplyr::select(all_of(c(genename_col, combined_pip_col, attributable_cols))) %>%
    pivot_longer(
      cols = all_of(attributable_cols),
      names_to = "Tissue",
      values_to = "Proportion"
    ) %>%
    mutate(Tissue = gsub("_Attributable_PIP \\(%\\)", "", Tissue))  # Modify Tissue names
  
  
  # Calculate the absolute values for each segment
  long_df_attributable <- long_df_attributable %>%
    mutate(Absolute_Value = Proportion / 100 * long_df_attributable[[combined_pip_col]])
  
  # Create the plot
  p <- ggplot(long_df_attributable, aes_string(x = genename_col, y = "Absolute_Value", fill = "Tissue")) +
    geom_bar(stat = "identity") +
    theme_minimal() +
    labs(title = "Stacked Histogram of Combined PIP with Tissue Contributions",
         x = "Gene Name",
         y = "Combined PIP",
         fill = "Tissue") +
    theme(axis.text.x = element_text(angle = 45, hjust = 1))
  
  return(p)
}

add_eqtl <- function(df_gene) {
  # Merge with gene_ensbl_symbl if needed
  # df_gene <- merge(df_gene, gene_ensbl_symbl, by="genename")
  df_gene <- merge(df_gene, gene_ensbl_symbl, by="genename")
  df_gene_extended <- df_gene
  candi_col <- seq(3, ncol(df_gene) - 1, by = 2)
  
  for (j in candi_col[1:5]) {
    tissue <- names(df_gene)[j]
    weight <- paste0("/project2/xinhe/shared_data/multigroup_ctwas/weights/predictdb_nolnc/mashr_", tissue, ".db")
    
    sqlite <- DBI::dbDriver("SQLite")
    db <- DBI::dbConnect(sqlite, weight)
    query <- function(...) DBI::dbGetQuery(db, ...)
    weight_info <- query("SELECT * FROM weights")
    DBI::dbDisconnect(db)
    
    weight_info <- weight_info[weight_info$gene %in% df_gene$gene,]
    norsid <- grep(pattern = "rs",x=weight_info$rsid,invert = T)
    weight_info$rsid[norsid] <- "norsid"
    
    rearranged_df <- weight_info %>%
      dplyr::group_by(gene) %>%
      dplyr::summarize(
        combined_info = paste(rsid, varID, round(weight, digits = 4), sep = ".", collapse = ", ")
      )
    
    merged_df <- dplyr::left_join(df_gene_extended, rearranged_df, by = "gene")
    new_col_name <- paste0("eqtl_pos_weight_for_", tissue)
    
    # Rename the column
    merged_df <- merged_df %>%
      dplyr::rename(!!new_col_name := combined_info)
    
    # Move the new column next to its corresponding tissue column
    tissue_position <- which(names(merged_df) == tissue)
    merged_df <- merged_df %>%
      dplyr::select(names(merged_df)[1:tissue_position], new_col_name, everything())
    
    # Update df_gene_extended for the next iteration
    df_gene_extended <- merged_df
  }
  
  return(df_gene_extended)
}

create_pie_chart_unexplained_added <- function(para) {
  # Extract relevant columns
  data <- para[, c("type", "group_pve")]
  
  # Calculate the 'unexplained' part
  total_pve <- sum(as.numeric(data$group_pve), na.rm = TRUE)
  unexplained_pve <- 1 - total_pve
  data <- rbind(data, data.frame(type = "Unexplained", group_pve = unexplained_pve))
  data$group_pve <- round(as.numeric(data$group_pve), digits = 5)
  
  # Set colors for each type, with specific colors for "Unexplained" and "SNP"
  unique_types <- unique(data$type)
  colors <- c("#e41a1c", "#377eb8", "#4daf4a", "#984ea3", "#ff7f00", "#ffff33", "#a65628")
  names(colors) <- data$type
  colors["Unexplained"] <- "grey"
  colors["SNP"] <- "#f781bf"
  
  # Adjust type labels to include group_pve values
  data$type <- paste0(data$type, ":", data$group_pve)
  names(colors) <- data$type
  
  # Create and display the pie chart
  pie <- ggplot(data, aes(x = "", y = group_pve, fill = type)) + 
    geom_bar(stat = "identity", width = 1) +
    coord_polar("y", start = 0) +
    scale_fill_manual(values = colors) +
    theme_void() +
    labs(fill = "Type") +
    ggtitle("Pie Chart of Group PVE")
  
  print(pie)
}

create_pie_chart <- function(para) {
  # Extract relevant columns
  data <- para[, c("type", "group_pve")]
  
  data$group_pve <- round(as.numeric(data$group_pve), digits = 5)
  
  # Set colors for each type, with specific colors for "Unexplained" and "SNP"
  unique_types <- unique(data$type)
  colors <- c("#e41a1c", "#377eb8", "#4daf4a", "#984ea3", "#ff7f00", "#ffff33")
  names(colors) <- data$type
  colors["SNP"] <- "grey"
  
  # Adjust type labels to include group_pve values
  data$type <- paste0(data$type, ":", data$group_pve)
  names(colors) <- data$type
  
  # Create and display the pie chart
  pie <- ggplot(data, aes(x = "", y = group_pve, fill = type)) + 
    geom_bar(stat = "identity", width = 1) +
    coord_polar("y", start = 0) +
    scale_fill_manual(values = colors) +
    theme_void() +
    labs(fill = "Type") +
    ggtitle("Pie Chart of Group PVE")
  
  print(pie)
}

create_pie_chart_relative_pve <- function(para) {
  
  data <- para[, c("type", "group_pve")]
  # Convert group_pve to numeric and calculate relative PVE
  data$group_pve <- as.numeric(data$group_pve)
  total_pve <- sum(data$group_pve, na.rm = TRUE)
  data$relative_pve <- round(data$group_pve / total_pve, digits = 5)

# Set colors for each type, with specific colors for "Unexplained" and "SNP"
  unique_types <- unique(data$type)
  colors <- c("#e41a1c", "#377eb8", "#4daf4a", "#984ea3", "#ff7f00", "#ffff33")
  names(colors) <- data$type
  colors["SNP"] <- "grey"

# Adjust type labels to include relative_pve values
  data$type <- paste0(data$type, ":", data$relative_pve)
  names(colors) <- data$type

# Create and display the pie chart
  pie <- ggplot(data, aes(x = "", y = relative_pve, fill = type)) + 
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  scale_fill_manual(values = colors) +
  theme_void() +
  labs(fill = "Type") +
  ggtitle("Pie Chart of Relative Group PVE")

  print(pie)
  
}


```

# LB

```{r fig.height=6, fig.width=12,message=FALSE, warning=FALSE}
outputdir <- "/project/xinhe/xsun/ctwas/2.TIBETN_coding/output_multi/LB/"
outname <- "LB_ctwas"
gwas_n <- 966
weight <- "Cells_Cultured_fibroblasts,Lung,Esophagus_Gastroesophageal_Junction,Muscle_Skeletal,Skin_Not_Sun_Exposed_Suprapubic"

load(paste0(outputdir,outname,"_z_snp.Rd"))
load(paste0(outputdir,outname,"_z_gene.Rd"))

ctwas_parameters <- ctwas:::ctwas_summarize_parameters(outputdir = outputdir,
                                               outname = outname,
                                               gwas_n = gwas_n,
                                               thin = thin)

ctwas_parameters$convergence_plot

ctwas_res <- process_data(weight = weight,outputdir = outputdir,outname = outname,z_snp = z_snp,z_gene = z_gene)

para <- process_ctwas_results(ctwas_parameters = ctwas_parameters,ctwas_res = ctwas_res)
DT::datatable(para,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Parameters'))
create_pie_chart_relative_pve(para)
df_gene <- process_gene_data(ctwas_res = ctwas_res)
df_gene <- attributable_pip(df_gene, "combined_pip")
df_gene_full <- add_eqtl(df_gene)
DT::datatable(df_gene_full,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Genes with combined pip > 0.8, 0 means pip=0; NA means no pip reported by ctwas or no such gene in this tissue'),options = list(pageLength = 10) )

attributable_columns <- colnames(df_gene)[seq(4,12,by=2)]

plot_stacked_histogram(df_gene, "genename", "combined_pip", attributable_columns)

table_show <- ctwas_res[ctwas_res$genename %in% df_gene$genename,]
table_show <- rbind(table_show,ctwas_res[ctwas_res$type!="SNP" & ctwas_res$susie_pip > 0.8,])
table_show <- table_show[!duplicated(table_show$id),]
DT::datatable(table_show,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Detailed table for genes with susiepip > 0.8 & combined pip >0.8'),options = list(pageLength = 5))

```

# MC

```{r fig.height=6, fig.width=12,message=FALSE, warning=FALSE}
outputdir <- "/project/xinhe/xsun/ctwas/2.TIBETN_coding/output_multi/MC/"
outname <- "MC_ctwas"
gwas_n <- 966
weight <- "Colon_Sigmoid,Testis,Nerve_Tibial,Cells_Cultured_fibroblasts,Pituitary"

load(paste0(outputdir,outname,"_z_snp.Rd"))
load(paste0(outputdir,outname,"_z_gene.Rd"))

ctwas_parameters <- ctwas:::ctwas_summarize_parameters(outputdir = outputdir,
                                               outname = outname,
                                               gwas_n = gwas_n,
                                               thin = thin)

ctwas_parameters$convergence_plot

ctwas_res <- process_data(weight = weight,outputdir = outputdir,outname = outname,z_snp = z_snp,z_gene = z_gene)

para <- process_ctwas_results(ctwas_parameters = ctwas_parameters,ctwas_res = ctwas_res)
DT::datatable(para,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Parameters'))
create_pie_chart_relative_pve(para)
df_gene <- process_gene_data(ctwas_res = ctwas_res)
df_gene <- attributable_pip(df_gene, "combined_pip")
df_gene_full <- add_eqtl(df_gene)
DT::datatable(df_gene_full,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Genes with combined pip > 0.8, 0 means pip=0; NA means no pip reported by ctwas or no such gene in this tissue'),options = list(pageLength = 10) )

attributable_columns <- colnames(df_gene)[seq(4,12,by=2)]

plot_stacked_histogram(df_gene, "genename", "combined_pip", attributable_columns)

table_show <- ctwas_res[ctwas_res$genename %in% df_gene$genename,]
table_show <- rbind(table_show,ctwas_res[ctwas_res$type!="SNP" & ctwas_res$susie_pip > 0.8,])
table_show <- table_show[!duplicated(table_show$id),]
DT::datatable(table_show,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Detailed table for genes with susiepip > 0.8 & combined pip >0.8'),options = list(pageLength = 5))

```



# OxHb

```{r fig.height=6, fig.width=12,message=FALSE, warning=FALSE}
outputdir <- "/project/xinhe/xsun/ctwas/2.TIBETN_coding/output_multi/OxHb/"
outname <- "OxHb_ctwas"
gwas_n <- 966
weight <- "Heart_Left_Ventricle,Liver,Skin_Sun_Exposed_Lower_leg,Stomach,Cells_Cultured_fibroblasts"

load(paste0(outputdir,outname,"_z_snp.Rd"))
load(paste0(outputdir,outname,"_z_gene.Rd"))

ctwas_parameters <- ctwas:::ctwas_summarize_parameters(outputdir = outputdir,
                                               outname = outname,
                                               gwas_n = gwas_n,
                                               thin = thin)

ctwas_parameters$convergence_plot

ctwas_res <- process_data(weight = weight,outputdir = outputdir,outname = outname,z_snp = z_snp,z_gene = z_gene)

para <- process_ctwas_results(ctwas_parameters = ctwas_parameters,ctwas_res = ctwas_res)
DT::datatable(para,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Parameters'))
create_pie_chart_relative_pve(para)
df_gene <- process_gene_data(ctwas_res = ctwas_res)
df_gene <- attributable_pip(df_gene, "combined_pip")
df_gene_full <- add_eqtl(df_gene)
DT::datatable(df_gene_full,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Genes with combined pip > 0.8, 0 means pip=0; NA means no pip reported by ctwas or no such gene in this tissue'),options = list(pageLength = 10) )

attributable_columns <- colnames(df_gene)[seq(4,12,by=2)]

plot_stacked_histogram(df_gene, "genename", "combined_pip", attributable_columns)

table_show <- ctwas_res[ctwas_res$genename %in% df_gene$genename,]
table_show <- rbind(table_show,ctwas_res[ctwas_res$type!="SNP" & ctwas_res$susie_pip > 0.8,])
table_show <- table_show[!duplicated(table_show$id),]
DT::datatable(table_show,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Detailed table for genes with susiepip > 0.8 & combined pip >0.8'),options = list(pageLength = 5))

```


# Pulse

```{r fig.height=6, fig.width=12,message=FALSE, warning=FALSE}
outputdir <- "/project/xinhe/xsun/ctwas/2.TIBETN_coding/output_multi/Pulse/"
outname <- "Pulse_ctwas"
gwas_n <- 966
weight <- "Esophagus_Mucosa,Heart_Atrial_Appendage,Esophagus_Gastroesophageal_Junction,Skin_Sun_Exposed_Lower_leg,Stomach"

load(paste0(outputdir,outname,"_z_snp.Rd"))
load(paste0(outputdir,outname,"_z_gene.Rd"))

ctwas_parameters <- ctwas:::ctwas_summarize_parameters(outputdir = outputdir,
                                               outname = outname,
                                               gwas_n = gwas_n,
                                               thin = thin)

ctwas_parameters$convergence_plot

ctwas_res <- process_data(weight = weight,outputdir = outputdir,outname = outname,z_snp = z_snp,z_gene = z_gene)

para <- process_ctwas_results(ctwas_parameters = ctwas_parameters,ctwas_res = ctwas_res)
DT::datatable(para,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Parameters'))
create_pie_chart_relative_pve(para)
df_gene <- process_gene_data(ctwas_res = ctwas_res)
df_gene <- attributable_pip(df_gene, "combined_pip")
df_gene_full <- add_eqtl(df_gene)
DT::datatable(df_gene_full,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Genes with combined pip > 0.8, 0 means pip=0; NA means no pip reported by ctwas or no such gene in this tissue'),options = list(pageLength = 10) )

attributable_columns <- colnames(df_gene)[seq(4,12,by=2)]

plot_stacked_histogram(df_gene, "genename", "combined_pip", attributable_columns)

table_show <- ctwas_res[ctwas_res$genename %in% df_gene$genename,]
table_show <- rbind(table_show,ctwas_res[ctwas_res$type!="SNP" & ctwas_res$susie_pip > 0.8,])
table_show <- table_show[!duplicated(table_show$id),]
DT::datatable(table_show,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Detailed table for genes with susiepip > 0.8 & combined pip >0.8'),options = list(pageLength = 5))

source("/project/xinhe/xsun/ctwas/2.TIBETN_coding/locus_plot_pulse.R")
locus_plot(region_tag = "3_93",ctwas_res = ctwas_res)
locus_plot(region_tag = "1_54",ctwas_res = ctwas_res)
```
